// @alicloud/vod20170321/dist/models/AddEditingProjectMaterialsResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddEditingProjectMaterialsResponseBody = exports.AddEditingProjectMaterialsResponseBodyMaterialList = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class AddEditingProjectMaterialsResponseBodyMaterialList extends $dara.Model {
    static names() {
        return {
            cateId: 'CateId',
            cateName: 'CateName',
            coverURL: 'CoverURL',
            createTime: 'CreateTime',
            customerId: 'CustomerId',
            description: 'Description',
            duration: 'Duration',
            materialId: 'MaterialId',
            materialType: 'MaterialType',
            modifyTime: 'ModifyTime',
            size: 'Size',
            snapshots: 'Snapshots',
            spriteConfig: 'SpriteConfig',
            sprites: 'Sprites',
            status: 'Status',
            tags: 'Tags',
            title: 'Title',
        };
    }
    static types() {
        return {
            cateId: 'number',
            cateName: 'string',
            coverURL: 'string',
            createTime: 'string',
            customerId: 'number',
            description: 'string',
            duration: 'number',
            materialId: 'string',
            materialType: 'string',
            modifyTime: 'string',
            size: 'number',
            snapshots: { 'type': 'array', 'itemType': 'string' },
            spriteConfig: 'string',
            sprites: { 'type': 'array', 'itemType': 'string' },
            status: 'string',
            tags: 'string',
            title: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.snapshots)) {
            $dara.Model.validateArray(this.snapshots);
        }
        if (Array.isArray(this.sprites)) {
            $dara.Model.validateArray(this.sprites);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.AddEditingProjectMaterialsResponseBodyMaterialList = AddEditingProjectMaterialsResponseBodyMaterialList;
class AddEditingProjectMaterialsResponseBody extends $dara.Model {
    static names() {
        return {
            materialList: 'MaterialList',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            materialList: { 'type': 'array', 'itemType': AddEditingProjectMaterialsResponseBodyMaterialList },
            requestId: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.materialList)) {
            $dara.Model.validateArray(this.materialList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.AddEditingProjectMaterialsResponseBody = AddEditingProjectMaterialsResponseBody;
