// @alicloud/vod20170321/dist/models/DescribeVodUserVipsByDomainResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodUserVipsByDomainResponseBody = exports.DescribeVodUserVipsByDomainResponseBodyVips = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodUserVipsByDomainResponseBodyVips extends $dara.Model {
    static names() {
        return {
            vip: 'Vip',
        };
    }
    static types() {
        return {
            vip: { 'type': 'array', 'itemType': 'string' },
        };
    }
    validate() {
        if (Array.isArray(this.vip)) {
            $dara.Model.validateArray(this.vip);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodUserVipsByDomainResponseBodyVips = DescribeVodUserVipsByDomainResponseBodyVips;
class DescribeVodUserVipsByDomainResponseBody extends $dara.Model {
    static names() {
        return {
            domainName: 'DomainName',
            requestId: 'RequestId',
            vips: 'Vips',
        };
    }
    static types() {
        return {
            domainName: 'string',
            requestId: 'string',
            vips: DescribeVodUserVipsByDomainResponseBodyVips,
        };
    }
    validate() {
        if (this.vips && typeof this.vips.validate === 'function') {
            this.vips.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodUserVipsByDomainResponseBody = DescribeVodUserVipsByDomainResponseBody;
