// @alicloud/vod20170321/dist/models/DescribeVodUserDomainsResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodUserDomainsResponseBody = exports.DescribeVodUserDomainsResponseBodyDomains = exports.DescribeVodUserDomainsResponseBodyDomainsPageData = exports.DescribeVodUserDomainsResponseBodyDomainsPageDataSources = exports.DescribeVodUserDomainsResponseBodyDomainsPageDataSourcesSource = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodUserDomainsResponseBodyDomainsPageDataSourcesSource extends $dara.Model {
    static names() {
        return {
            content: 'Content',
            port: 'Port',
            priority: 'Priority',
            type: 'Type',
        };
    }
    static types() {
        return {
            content: 'string',
            port: 'number',
            priority: 'string',
            type: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodUserDomainsResponseBodyDomainsPageDataSourcesSource = DescribeVodUserDomainsResponseBodyDomainsPageDataSourcesSource;
class DescribeVodUserDomainsResponseBodyDomainsPageDataSources extends $dara.Model {
    static names() {
        return {
            source: 'Source',
        };
    }
    static types() {
        return {
            source: { 'type': 'array', 'itemType': DescribeVodUserDomainsResponseBodyDomainsPageDataSourcesSource },
        };
    }
    validate() {
        if (Array.isArray(this.source)) {
            $dara.Model.validateArray(this.source);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodUserDomainsResponseBodyDomainsPageDataSources = DescribeVodUserDomainsResponseBodyDomainsPageDataSources;
class DescribeVodUserDomainsResponseBodyDomainsPageData extends $dara.Model {
    static names() {
        return {
            cname: 'Cname',
            description: 'Description',
            domainName: 'DomainName',
            domainStatus: 'DomainStatus',
            gmtCreated: 'GmtCreated',
            gmtModified: 'GmtModified',
            sandbox: 'Sandbox',
            sources: 'Sources',
            sslProtocol: 'SslProtocol',
        };
    }
    static types() {
        return {
            cname: 'string',
            description: 'string',
            domainName: 'string',
            domainStatus: 'string',
            gmtCreated: 'string',
            gmtModified: 'string',
            sandbox: 'string',
            sources: DescribeVodUserDomainsResponseBodyDomainsPageDataSources,
            sslProtocol: 'string',
        };
    }
    validate() {
        if (this.sources && typeof this.sources.validate === 'function') {
            this.sources.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodUserDomainsResponseBodyDomainsPageData = DescribeVodUserDomainsResponseBodyDomainsPageData;
class DescribeVodUserDomainsResponseBodyDomains extends $dara.Model {
    static names() {
        return {
            pageData: 'PageData',
        };
    }
    static types() {
        return {
            pageData: { 'type': 'array', 'itemType': DescribeVodUserDomainsResponseBodyDomainsPageData },
        };
    }
    validate() {
        if (Array.isArray(this.pageData)) {
            $dara.Model.validateArray(this.pageData);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodUserDomainsResponseBodyDomains = DescribeVodUserDomainsResponseBodyDomains;
class DescribeVodUserDomainsResponseBody extends $dara.Model {
    static names() {
        return {
            domains: 'Domains',
            pageNumber: 'PageNumber',
            pageSize: 'PageSize',
            requestId: 'RequestId',
            totalCount: 'TotalCount',
        };
    }
    static types() {
        return {
            domains: DescribeVodUserDomainsResponseBodyDomains,
            pageNumber: 'number',
            pageSize: 'number',
            requestId: 'string',
            totalCount: 'number',
        };
    }
    validate() {
        if (this.domains && typeof this.domains.validate === 'function') {
            this.domains.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodUserDomainsResponseBody = DescribeVodUserDomainsResponseBody;
