// @alicloud/vod20170321/dist/models/DescribeVodRefreshTasksResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodRefreshTasksResponseBody = exports.DescribeVodRefreshTasksResponseBodyTasks = exports.DescribeVodRefreshTasksResponseBodyTasksTask = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodRefreshTasksResponseBodyTasksTask extends $dara.Model {
    static names() {
        return {
            creationTime: 'CreationTime',
            description: 'Description',
            objectPath: 'ObjectPath',
            objectType: 'ObjectType',
            process: 'Process',
            status: 'Status',
            taskId: 'TaskId',
        };
    }
    static types() {
        return {
            creationTime: 'string',
            description: 'string',
            objectPath: 'string',
            objectType: 'string',
            process: 'string',
            status: 'string',
            taskId: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodRefreshTasksResponseBodyTasksTask = DescribeVodRefreshTasksResponseBodyTasksTask;
class DescribeVodRefreshTasksResponseBodyTasks extends $dara.Model {
    static names() {
        return {
            task: 'Task',
        };
    }
    static types() {
        return {
            task: { 'type': 'array', 'itemType': DescribeVodRefreshTasksResponseBodyTasksTask },
        };
    }
    validate() {
        if (Array.isArray(this.task)) {
            $dara.Model.validateArray(this.task);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodRefreshTasksResponseBodyTasks = DescribeVodRefreshTasksResponseBodyTasks;
class DescribeVodRefreshTasksResponseBody extends $dara.Model {
    static names() {
        return {
            pageNumber: 'PageNumber',
            pageSize: 'PageSize',
            requestId: 'RequestId',
            tasks: 'Tasks',
            totalCount: 'TotalCount',
        };
    }
    static types() {
        return {
            pageNumber: 'number',
            pageSize: 'number',
            requestId: 'string',
            tasks: DescribeVodRefreshTasksResponseBodyTasks,
            totalCount: 'number',
        };
    }
    validate() {
        if (this.tasks && typeof this.tasks.validate === 'function') {
            this.tasks.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodRefreshTasksResponseBody = DescribeVodRefreshTasksResponseBody;
