// @alicloud/vod20170321/dist/models/DescribePlayTopVideosResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribePlayTopVideosResponseBody = exports.DescribePlayTopVideosResponseBodyTopPlayVideos = exports.DescribePlayTopVideosResponseBodyTopPlayVideosTopPlayVideoStatis = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribePlayTopVideosResponseBodyTopPlayVideosTopPlayVideoStatis extends $dara.Model {
    static names() {
        return {
            playDuration: 'PlayDuration',
            title: 'Title',
            UV: 'UV',
            VV: 'VV',
            videoId: 'VideoId',
        };
    }
    static types() {
        return {
            playDuration: 'string',
            title: 'string',
            UV: 'string',
            VV: 'string',
            videoId: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribePlayTopVideosResponseBodyTopPlayVideosTopPlayVideoStatis = DescribePlayTopVideosResponseBodyTopPlayVideosTopPlayVideoStatis;
class DescribePlayTopVideosResponseBodyTopPlayVideos extends $dara.Model {
    static names() {
        return {
            topPlayVideoStatis: 'TopPlayVideoStatis',
        };
    }
    static types() {
        return {
            topPlayVideoStatis: { 'type': 'array', 'itemType': DescribePlayTopVideosResponseBodyTopPlayVideosTopPlayVideoStatis },
        };
    }
    validate() {
        if (Array.isArray(this.topPlayVideoStatis)) {
            $dara.Model.validateArray(this.topPlayVideoStatis);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribePlayTopVideosResponseBodyTopPlayVideos = DescribePlayTopVideosResponseBodyTopPlayVideos;
class DescribePlayTopVideosResponseBody extends $dara.Model {
    static names() {
        return {
            pageNo: 'PageNo',
            pageSize: 'PageSize',
            requestId: 'RequestId',
            topPlayVideos: 'TopPlayVideos',
            totalNum: 'TotalNum',
        };
    }
    static types() {
        return {
            pageNo: 'number',
            pageSize: 'number',
            requestId: 'string',
            topPlayVideos: DescribePlayTopVideosResponseBodyTopPlayVideos,
            totalNum: 'number',
        };
    }
    validate() {
        if (this.topPlayVideos && typeof this.topPlayVideos.validate === 'function') {
            this.topPlayVideos.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribePlayTopVideosResponseBody = DescribePlayTopVideosResponseBody;
