// @alicloud/vod20170321/dist/models/DescribeVodCertificateListResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodCertificateListResponseBody = exports.DescribeVodCertificateListResponseBodyCertificateListModel = exports.DescribeVodCertificateListResponseBodyCertificateListModelCertList = exports.DescribeVodCertificateListResponseBodyCertificateListModelCertListCert = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodCertificateListResponseBodyCertificateListModelCertListCert extends $dara.Model {
    static names() {
        return {
            algorithm: 'Algorithm',
            certId: 'CertId',
            certIdentifier: 'CertIdentifier',
            certName: 'CertName',
            common: 'Common',
            createTime: 'CreateTime',
            domainMatchCert: 'DomainMatchCert',
            endTime: 'EndTime',
            fingerprint: 'Fingerprint',
            instanceId: 'InstanceId',
            issuer: 'Issuer',
            lastTime: 'LastTime',
            signAlgorithm: 'SignAlgorithm',
        };
    }
    static types() {
        return {
            algorithm: 'string',
            certId: 'number',
            certIdentifier: 'string',
            certName: 'string',
            common: 'string',
            createTime: 'number',
            domainMatchCert: 'boolean',
            endTime: 'number',
            fingerprint: 'string',
            instanceId: 'string',
            issuer: 'string',
            lastTime: 'number',
            signAlgorithm: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodCertificateListResponseBodyCertificateListModelCertListCert = DescribeVodCertificateListResponseBodyCertificateListModelCertListCert;
class DescribeVodCertificateListResponseBodyCertificateListModelCertList extends $dara.Model {
    static names() {
        return {
            cert: 'Cert',
        };
    }
    static types() {
        return {
            cert: { 'type': 'array', 'itemType': DescribeVodCertificateListResponseBodyCertificateListModelCertListCert },
        };
    }
    validate() {
        if (Array.isArray(this.cert)) {
            $dara.Model.validateArray(this.cert);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodCertificateListResponseBodyCertificateListModelCertList = DescribeVodCertificateListResponseBodyCertificateListModelCertList;
class DescribeVodCertificateListResponseBodyCertificateListModel extends $dara.Model {
    static names() {
        return {
            certList: 'CertList',
            count: 'Count',
        };
    }
    static types() {
        return {
            certList: DescribeVodCertificateListResponseBodyCertificateListModelCertList,
            count: 'number',
        };
    }
    validate() {
        if (this.certList && typeof this.certList.validate === 'function') {
            this.certList.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodCertificateListResponseBodyCertificateListModel = DescribeVodCertificateListResponseBodyCertificateListModel;
class DescribeVodCertificateListResponseBody extends $dara.Model {
    static names() {
        return {
            certificateListModel: 'CertificateListModel',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            certificateListModel: DescribeVodCertificateListResponseBodyCertificateListModel,
            requestId: 'string',
        };
    }
    validate() {
        if (this.certificateListModel && typeof this.certificateListModel.validate === 'function') {
            this.certificateListModel.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodCertificateListResponseBody = DescribeVodCertificateListResponseBody;
