// @alicloud/vod20170321/dist/models/DescribeVodMediaPlayDataResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodMediaPlayDataResponseBody = exports.DescribeVodMediaPlayDataResponseBodyQoeInfoList = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodMediaPlayDataResponseBodyQoeInfoList extends $dara.Model {
    static names() {
        return {
            DAU: 'DAU',
            mediaId: 'MediaId',
            playDuration: 'PlayDuration',
            playDurationPerUv: 'PlayDurationPerUv',
            playPerVv: 'PlayPerVv',
            playSuccessVv: 'PlaySuccessVv',
            videoDuration: 'VideoDuration',
            videoTitle: 'VideoTitle',
        };
    }
    static types() {
        return {
            DAU: 'number',
            mediaId: 'string',
            playDuration: 'number',
            playDurationPerUv: 'number',
            playPerVv: 'number',
            playSuccessVv: 'number',
            videoDuration: 'number',
            videoTitle: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodMediaPlayDataResponseBodyQoeInfoList = DescribeVodMediaPlayDataResponseBodyQoeInfoList;
class DescribeVodMediaPlayDataResponseBody extends $dara.Model {
    static names() {
        return {
            pageNo: 'PageNo',
            pageSize: 'PageSize',
            qoeInfoList: 'QoeInfoList',
            requestId: 'RequestId',
            totalCount: 'TotalCount',
        };
    }
    static types() {
        return {
            pageNo: 'number',
            pageSize: 'number',
            qoeInfoList: { 'type': 'array', 'itemType': DescribeVodMediaPlayDataResponseBodyQoeInfoList },
            requestId: 'string',
            totalCount: 'number',
        };
    }
    validate() {
        if (Array.isArray(this.qoeInfoList)) {
            $dara.Model.validateArray(this.qoeInfoList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodMediaPlayDataResponseBody = DescribeVodMediaPlayDataResponseBody;
