// @alicloud/vod20170321/dist/models/GetAimediaAuditJobResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAIMediaAuditJobResponseBody = exports.GetAIMediaAuditJobResponseBodyMediaAuditJob = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobData = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResult = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResult = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResultTopList = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResultCounterList = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResult = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResultTopList = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResultCounterList = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResult = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResultTopList = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResultCounterList = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResult = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResultTopList = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResultCounterList = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResult = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResultTopList = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResultCounterList = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataTextResult = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataImageResult = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataImageResultResult = exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataAudioResult = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataAudioResult extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            scene: 'Scene',
            score: 'Score',
            suggestion: 'Suggestion',
        };
    }
    static types() {
        return {
            label: 'string',
            scene: 'string',
            score: 'string',
            suggestion: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataAudioResult = GetAIMediaAuditJobResponseBodyMediaAuditJobDataAudioResult;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataImageResultResult extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            scene: 'Scene',
            score: 'Score',
            suggestion: 'Suggestion',
        };
    }
    static types() {
        return {
            label: 'string',
            scene: 'string',
            score: 'string',
            suggestion: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataImageResultResult = GetAIMediaAuditJobResponseBodyMediaAuditJobDataImageResultResult;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataImageResult extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            result: 'Result',
            suggestion: 'Suggestion',
            type: 'Type',
            url: 'Url',
        };
    }
    static types() {
        return {
            label: 'string',
            result: { 'type': 'array', 'itemType': GetAIMediaAuditJobResponseBodyMediaAuditJobDataImageResultResult },
            suggestion: 'string',
            type: 'string',
            url: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.result)) {
            $dara.Model.validateArray(this.result);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataImageResult = GetAIMediaAuditJobResponseBodyMediaAuditJobDataImageResult;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataTextResult extends $dara.Model {
    static names() {
        return {
            content: 'Content',
            label: 'Label',
            scene: 'Scene',
            score: 'Score',
            suggestion: 'Suggestion',
            type: 'Type',
        };
    }
    static types() {
        return {
            content: 'string',
            label: 'string',
            scene: 'string',
            score: 'string',
            suggestion: 'string',
            type: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataTextResult = GetAIMediaAuditJobResponseBodyMediaAuditJobDataTextResult;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResultCounterList extends $dara.Model {
    static names() {
        return {
            count: 'Count',
            label: 'Label',
        };
    }
    static types() {
        return {
            count: 'number',
            label: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResultCounterList = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResultCounterList;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResultTopList extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
            url: 'Url',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
            url: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResultTopList = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResultTopList;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResult extends $dara.Model {
    static names() {
        return {
            averageScore: 'AverageScore',
            counterList: 'CounterList',
            label: 'Label',
            maxScore: 'MaxScore',
            suggestion: 'Suggestion',
            topList: 'TopList',
        };
    }
    static types() {
        return {
            averageScore: 'string',
            counterList: { 'type': 'array', 'itemType': GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResultCounterList },
            label: 'string',
            maxScore: 'string',
            suggestion: 'string',
            topList: { 'type': 'array', 'itemType': GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResultTopList },
        };
    }
    validate() {
        if (Array.isArray(this.counterList)) {
            $dara.Model.validateArray(this.counterList);
        }
        if (Array.isArray(this.topList)) {
            $dara.Model.validateArray(this.topList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResult = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResult;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResultCounterList extends $dara.Model {
    static names() {
        return {
            count: 'Count',
            label: 'Label',
        };
    }
    static types() {
        return {
            count: 'number',
            label: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResultCounterList = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResultCounterList;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResultTopList extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
            url: 'Url',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
            url: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResultTopList = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResultTopList;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResult extends $dara.Model {
    static names() {
        return {
            averageScore: 'AverageScore',
            counterList: 'CounterList',
            label: 'Label',
            maxScore: 'MaxScore',
            suggestion: 'Suggestion',
            topList: 'TopList',
        };
    }
    static types() {
        return {
            averageScore: 'string',
            counterList: { 'type': 'array', 'itemType': GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResultCounterList },
            label: 'string',
            maxScore: 'string',
            suggestion: 'string',
            topList: { 'type': 'array', 'itemType': GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResultTopList },
        };
    }
    validate() {
        if (Array.isArray(this.counterList)) {
            $dara.Model.validateArray(this.counterList);
        }
        if (Array.isArray(this.topList)) {
            $dara.Model.validateArray(this.topList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResult = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResult;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResultCounterList extends $dara.Model {
    static names() {
        return {
            count: 'Count',
            label: 'Label',
        };
    }
    static types() {
        return {
            count: 'number',
            label: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResultCounterList = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResultCounterList;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResultTopList extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
            url: 'Url',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
            url: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResultTopList = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResultTopList;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResult extends $dara.Model {
    static names() {
        return {
            averageScore: 'AverageScore',
            counterList: 'CounterList',
            label: 'Label',
            maxScore: 'MaxScore',
            suggestion: 'Suggestion',
            topList: 'TopList',
        };
    }
    static types() {
        return {
            averageScore: 'string',
            counterList: { 'type': 'array', 'itemType': GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResultCounterList },
            label: 'string',
            maxScore: 'string',
            suggestion: 'string',
            topList: { 'type': 'array', 'itemType': GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResultTopList },
        };
    }
    validate() {
        if (Array.isArray(this.counterList)) {
            $dara.Model.validateArray(this.counterList);
        }
        if (Array.isArray(this.topList)) {
            $dara.Model.validateArray(this.topList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResult = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResult;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResultCounterList extends $dara.Model {
    static names() {
        return {
            count: 'Count',
            label: 'Label',
        };
    }
    static types() {
        return {
            count: 'number',
            label: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResultCounterList = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResultCounterList;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResultTopList extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
            url: 'Url',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
            url: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResultTopList = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResultTopList;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResult extends $dara.Model {
    static names() {
        return {
            averageScore: 'AverageScore',
            counterList: 'CounterList',
            label: 'Label',
            maxScore: 'MaxScore',
            suggestion: 'Suggestion',
            topList: 'TopList',
        };
    }
    static types() {
        return {
            averageScore: 'string',
            counterList: { 'type': 'array', 'itemType': GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResultCounterList },
            label: 'string',
            maxScore: 'string',
            suggestion: 'string',
            topList: { 'type': 'array', 'itemType': GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResultTopList },
        };
    }
    validate() {
        if (Array.isArray(this.counterList)) {
            $dara.Model.validateArray(this.counterList);
        }
        if (Array.isArray(this.topList)) {
            $dara.Model.validateArray(this.topList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResult = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResult;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResultCounterList extends $dara.Model {
    static names() {
        return {
            count: 'Count',
            label: 'Label',
        };
    }
    static types() {
        return {
            count: 'number',
            label: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResultCounterList = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResultCounterList;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResultTopList extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
            url: 'Url',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
            url: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResultTopList = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResultTopList;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResult extends $dara.Model {
    static names() {
        return {
            averageScore: 'AverageScore',
            counterList: 'CounterList',
            label: 'Label',
            maxScore: 'MaxScore',
            suggestion: 'Suggestion',
            topList: 'TopList',
        };
    }
    static types() {
        return {
            averageScore: 'string',
            counterList: { 'type': 'array', 'itemType': GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResultCounterList },
            label: 'string',
            maxScore: 'string',
            suggestion: 'string',
            topList: { 'type': 'array', 'itemType': GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResultTopList },
        };
    }
    validate() {
        if (Array.isArray(this.counterList)) {
            $dara.Model.validateArray(this.counterList);
        }
        if (Array.isArray(this.topList)) {
            $dara.Model.validateArray(this.topList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResult = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResult;
class GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResult extends $dara.Model {
    static names() {
        return {
            adResult: 'AdResult',
            label: 'Label',
            liveResult: 'LiveResult',
            logoResult: 'LogoResult',
            pornResult: 'PornResult',
            suggestion: 'Suggestion',
            terrorismResult: 'TerrorismResult',
        };
    }
    static types() {
        return {
            adResult: GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultAdResult,
            label: 'string',
            liveResult: GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLiveResult,
            logoResult: GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultLogoResult,
            pornResult: GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultPornResult,
            suggestion: 'string',
            terrorismResult: GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResultTerrorismResult,
        };
    }
    validate() {
        if (this.adResult && typeof this.adResult.validate === 'function') {
            this.adResult.validate();
        }
        if (this.liveResult && typeof this.liveResult.validate === 'function') {
            this.liveResult.validate();
        }
        if (this.logoResult && typeof this.logoResult.validate === 'function') {
            this.logoResult.validate();
        }
        if (this.pornResult && typeof this.pornResult.validate === 'function') {
            this.pornResult.validate();
        }
        if (this.terrorismResult && typeof this.terrorismResult.validate === 'function') {
            this.terrorismResult.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResult = GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResult;
class GetAIMediaAuditJobResponseBodyMediaAuditJobData extends $dara.Model {
    static names() {
        return {
            abnormalModules: 'AbnormalModules',
            audioResult: 'AudioResult',
            imageResult: 'ImageResult',
            label: 'Label',
            suggestion: 'Suggestion',
            textResult: 'TextResult',
            videoResult: 'VideoResult',
        };
    }
    static types() {
        return {
            abnormalModules: 'string',
            audioResult: { 'type': 'array', 'itemType': GetAIMediaAuditJobResponseBodyMediaAuditJobDataAudioResult },
            imageResult: { 'type': 'array', 'itemType': GetAIMediaAuditJobResponseBodyMediaAuditJobDataImageResult },
            label: 'string',
            suggestion: 'string',
            textResult: { 'type': 'array', 'itemType': GetAIMediaAuditJobResponseBodyMediaAuditJobDataTextResult },
            videoResult: GetAIMediaAuditJobResponseBodyMediaAuditJobDataVideoResult,
        };
    }
    validate() {
        if (Array.isArray(this.audioResult)) {
            $dara.Model.validateArray(this.audioResult);
        }
        if (Array.isArray(this.imageResult)) {
            $dara.Model.validateArray(this.imageResult);
        }
        if (Array.isArray(this.textResult)) {
            $dara.Model.validateArray(this.textResult);
        }
        if (this.videoResult && typeof this.videoResult.validate === 'function') {
            this.videoResult.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJobData = GetAIMediaAuditJobResponseBodyMediaAuditJobData;
class GetAIMediaAuditJobResponseBodyMediaAuditJob extends $dara.Model {
    static names() {
        return {
            code: 'Code',
            completeTime: 'CompleteTime',
            creationTime: 'CreationTime',
            data: 'Data',
            jobId: 'JobId',
            mediaId: 'MediaId',
            message: 'Message',
            status: 'Status',
            type: 'Type',
        };
    }
    static types() {
        return {
            code: 'string',
            completeTime: 'string',
            creationTime: 'string',
            data: GetAIMediaAuditJobResponseBodyMediaAuditJobData,
            jobId: 'string',
            mediaId: 'string',
            message: 'string',
            status: 'string',
            type: 'string',
        };
    }
    validate() {
        if (this.data && typeof this.data.validate === 'function') {
            this.data.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBodyMediaAuditJob = GetAIMediaAuditJobResponseBodyMediaAuditJob;
class GetAIMediaAuditJobResponseBody extends $dara.Model {
    static names() {
        return {
            mediaAuditJob: 'MediaAuditJob',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            mediaAuditJob: GetAIMediaAuditJobResponseBodyMediaAuditJob,
            requestId: 'string',
        };
    }
    validate() {
        if (this.mediaAuditJob && typeof this.mediaAuditJob.validate === 'function') {
            this.mediaAuditJob.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIMediaAuditJobResponseBody = GetAIMediaAuditJobResponseBody;
