// @alicloud/vod20170321/dist/models/GetAttachedMediaInfoResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAttachedMediaInfoResponseBody = exports.GetAttachedMediaInfoResponseBodyAttachedMediaList = exports.GetAttachedMediaInfoResponseBodyAttachedMediaListCategories = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetAttachedMediaInfoResponseBodyAttachedMediaListCategories extends $dara.Model {
    static names() {
        return {
            cateId: 'CateId',
            cateName: 'CateName',
            level: 'Level',
            parentId: 'ParentId',
        };
    }
    static types() {
        return {
            cateId: 'number',
            cateName: 'string',
            level: 'number',
            parentId: 'number',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAttachedMediaInfoResponseBodyAttachedMediaListCategories = GetAttachedMediaInfoResponseBodyAttachedMediaListCategories;
class GetAttachedMediaInfoResponseBodyAttachedMediaList extends $dara.Model {
    static names() {
        return {
            appId: 'AppId',
            categories: 'Categories',
            creationTime: 'CreationTime',
            description: 'Description',
            mediaId: 'MediaId',
            modificationTime: 'ModificationTime',
            status: 'Status',
            storageLocation: 'StorageLocation',
            tags: 'Tags',
            title: 'Title',
            type: 'Type',
            URL: 'URL',
        };
    }
    static types() {
        return {
            appId: 'string',
            categories: { 'type': 'array', 'itemType': GetAttachedMediaInfoResponseBodyAttachedMediaListCategories },
            creationTime: 'string',
            description: 'string',
            mediaId: 'string',
            modificationTime: 'string',
            status: 'string',
            storageLocation: 'string',
            tags: 'string',
            title: 'string',
            type: 'string',
            URL: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.categories)) {
            $dara.Model.validateArray(this.categories);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAttachedMediaInfoResponseBodyAttachedMediaList = GetAttachedMediaInfoResponseBodyAttachedMediaList;
class GetAttachedMediaInfoResponseBody extends $dara.Model {
    static names() {
        return {
            attachedMediaList: 'AttachedMediaList',
            nonExistMediaIds: 'NonExistMediaIds',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            attachedMediaList: { 'type': 'array', 'itemType': GetAttachedMediaInfoResponseBodyAttachedMediaList },
            nonExistMediaIds: { 'type': 'array', 'itemType': 'string' },
            requestId: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.attachedMediaList)) {
            $dara.Model.validateArray(this.attachedMediaList);
        }
        if (Array.isArray(this.nonExistMediaIds)) {
            $dara.Model.validateArray(this.nonExistMediaIds);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAttachedMediaInfoResponseBody = GetAttachedMediaInfoResponseBody;
