// @alicloud/vod20170321/dist/models/GetAivideoTagResultResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAIVideoTagResultResponseBody = exports.GetAIVideoTagResultResponseBodyVideoTagResult = exports.GetAIVideoTagResultResponseBodyVideoTagResultTime = exports.GetAIVideoTagResultResponseBodyVideoTagResultPerson = exports.GetAIVideoTagResultResponseBodyVideoTagResultLocation = exports.GetAIVideoTagResultResponseBodyVideoTagResultKeyword = exports.GetAIVideoTagResultResponseBodyVideoTagResultCategory = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetAIVideoTagResultResponseBodyVideoTagResultCategory extends $dara.Model {
    static names() {
        return {
            tag: 'Tag',
        };
    }
    static types() {
        return {
            tag: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIVideoTagResultResponseBodyVideoTagResultCategory = GetAIVideoTagResultResponseBodyVideoTagResultCategory;
class GetAIVideoTagResultResponseBodyVideoTagResultKeyword extends $dara.Model {
    static names() {
        return {
            tag: 'Tag',
            times: 'Times',
        };
    }
    static types() {
        return {
            tag: 'string',
            times: { 'type': 'array', 'itemType': 'string' },
        };
    }
    validate() {
        if (Array.isArray(this.times)) {
            $dara.Model.validateArray(this.times);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIVideoTagResultResponseBodyVideoTagResultKeyword = GetAIVideoTagResultResponseBodyVideoTagResultKeyword;
class GetAIVideoTagResultResponseBodyVideoTagResultLocation extends $dara.Model {
    static names() {
        return {
            tag: 'Tag',
            times: 'Times',
        };
    }
    static types() {
        return {
            tag: 'string',
            times: { 'type': 'array', 'itemType': 'string' },
        };
    }
    validate() {
        if (Array.isArray(this.times)) {
            $dara.Model.validateArray(this.times);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIVideoTagResultResponseBodyVideoTagResultLocation = GetAIVideoTagResultResponseBodyVideoTagResultLocation;
class GetAIVideoTagResultResponseBodyVideoTagResultPerson extends $dara.Model {
    static names() {
        return {
            faceUrl: 'FaceUrl',
            tag: 'Tag',
            times: 'Times',
        };
    }
    static types() {
        return {
            faceUrl: 'string',
            tag: 'string',
            times: { 'type': 'array', 'itemType': 'string' },
        };
    }
    validate() {
        if (Array.isArray(this.times)) {
            $dara.Model.validateArray(this.times);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIVideoTagResultResponseBodyVideoTagResultPerson = GetAIVideoTagResultResponseBodyVideoTagResultPerson;
class GetAIVideoTagResultResponseBodyVideoTagResultTime extends $dara.Model {
    static names() {
        return {
            tag: 'Tag',
            times: 'Times',
        };
    }
    static types() {
        return {
            tag: 'string',
            times: { 'type': 'array', 'itemType': 'string' },
        };
    }
    validate() {
        if (Array.isArray(this.times)) {
            $dara.Model.validateArray(this.times);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIVideoTagResultResponseBodyVideoTagResultTime = GetAIVideoTagResultResponseBodyVideoTagResultTime;
class GetAIVideoTagResultResponseBodyVideoTagResult extends $dara.Model {
    static names() {
        return {
            category: 'Category',
            keyword: 'Keyword',
            location: 'Location',
            person: 'Person',
            time: 'Time',
        };
    }
    static types() {
        return {
            category: { 'type': 'array', 'itemType': GetAIVideoTagResultResponseBodyVideoTagResultCategory },
            keyword: { 'type': 'array', 'itemType': GetAIVideoTagResultResponseBodyVideoTagResultKeyword },
            location: { 'type': 'array', 'itemType': GetAIVideoTagResultResponseBodyVideoTagResultLocation },
            person: { 'type': 'array', 'itemType': GetAIVideoTagResultResponseBodyVideoTagResultPerson },
            time: { 'type': 'array', 'itemType': GetAIVideoTagResultResponseBodyVideoTagResultTime },
        };
    }
    validate() {
        if (Array.isArray(this.category)) {
            $dara.Model.validateArray(this.category);
        }
        if (Array.isArray(this.keyword)) {
            $dara.Model.validateArray(this.keyword);
        }
        if (Array.isArray(this.location)) {
            $dara.Model.validateArray(this.location);
        }
        if (Array.isArray(this.person)) {
            $dara.Model.validateArray(this.person);
        }
        if (Array.isArray(this.time)) {
            $dara.Model.validateArray(this.time);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIVideoTagResultResponseBodyVideoTagResult = GetAIVideoTagResultResponseBodyVideoTagResult;
class GetAIVideoTagResultResponseBody extends $dara.Model {
    static names() {
        return {
            requestId: 'RequestId',
            videoTagResult: 'VideoTagResult',
        };
    }
    static types() {
        return {
            requestId: 'string',
            videoTagResult: GetAIVideoTagResultResponseBodyVideoTagResult,
        };
    }
    validate() {
        if (this.videoTagResult && typeof this.videoTagResult.validate === 'function') {
            this.videoTagResult.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetAIVideoTagResultResponseBody = GetAIVideoTagResultResponseBody;
