// @alicloud/vod20170321/dist/models/GetMediaRefreshJobsResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetMediaRefreshJobsResponseBody = exports.GetMediaRefreshJobsResponseBodyMediaRefreshJobs = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetMediaRefreshJobsResponseBodyMediaRefreshJobs extends $dara.Model {
    static names() {
        return {
            errorCode: 'ErrorCode',
            errorMessage: 'ErrorMessage',
            filterPolicy: 'FilterPolicy',
            gmtCreate: 'GmtCreate',
            gmtModified: 'GmtModified',
            mediaId: 'MediaId',
            mediaRefreshJobId: 'MediaRefreshJobId',
            status: 'Status',
            successPlayUrls: 'SuccessPlayUrls',
            taskIds: 'TaskIds',
            taskType: 'TaskType',
            userData: 'UserData',
        };
    }
    static types() {
        return {
            errorCode: 'string',
            errorMessage: 'string',
            filterPolicy: 'string',
            gmtCreate: 'string',
            gmtModified: 'string',
            mediaId: 'string',
            mediaRefreshJobId: 'string',
            status: 'string',
            successPlayUrls: 'string',
            taskIds: 'string',
            taskType: 'string',
            userData: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaRefreshJobsResponseBodyMediaRefreshJobs = GetMediaRefreshJobsResponseBodyMediaRefreshJobs;
class GetMediaRefreshJobsResponseBody extends $dara.Model {
    static names() {
        return {
            mediaRefreshJobs: 'MediaRefreshJobs',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            mediaRefreshJobs: { 'type': 'array', 'itemType': GetMediaRefreshJobsResponseBodyMediaRefreshJobs },
            requestId: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.mediaRefreshJobs)) {
            $dara.Model.validateArray(this.mediaRefreshJobs);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaRefreshJobsResponseBody = GetMediaRefreshJobsResponseBody;
