// @alicloud/vod20170321/dist/models/GetMediaAuditResultDetailResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetMediaAuditResultDetailResponseBody = exports.GetMediaAuditResultDetailResponseBodyMediaAuditResultDetail = exports.GetMediaAuditResultDetailResponseBodyMediaAuditResultDetailList = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetMediaAuditResultDetailResponseBodyMediaAuditResultDetailList extends $dara.Model {
    static names() {
        return {
            adLabel: 'AdLabel',
            adScore: 'AdScore',
            liveLabel: 'LiveLabel',
            liveScore: 'LiveScore',
            logoLabel: 'LogoLabel',
            logoScore: 'LogoScore',
            pornLabel: 'PornLabel',
            pornScore: 'PornScore',
            terrorismLabel: 'TerrorismLabel',
            terrorismScore: 'TerrorismScore',
            timestamp: 'Timestamp',
            url: 'Url',
        };
    }
    static types() {
        return {
            adLabel: 'string',
            adScore: 'string',
            liveLabel: 'string',
            liveScore: 'string',
            logoLabel: 'string',
            logoScore: 'string',
            pornLabel: 'string',
            pornScore: 'string',
            terrorismLabel: 'string',
            terrorismScore: 'string',
            timestamp: 'string',
            url: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultDetailResponseBodyMediaAuditResultDetailList = GetMediaAuditResultDetailResponseBodyMediaAuditResultDetailList;
class GetMediaAuditResultDetailResponseBodyMediaAuditResultDetail extends $dara.Model {
    static names() {
        return {
            list: 'List',
            total: 'Total',
        };
    }
    static types() {
        return {
            list: { 'type': 'array', 'itemType': GetMediaAuditResultDetailResponseBodyMediaAuditResultDetailList },
            total: 'number',
        };
    }
    validate() {
        if (Array.isArray(this.list)) {
            $dara.Model.validateArray(this.list);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultDetailResponseBodyMediaAuditResultDetail = GetMediaAuditResultDetailResponseBodyMediaAuditResultDetail;
class GetMediaAuditResultDetailResponseBody extends $dara.Model {
    static names() {
        return {
            mediaAuditResultDetail: 'MediaAuditResultDetail',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            mediaAuditResultDetail: GetMediaAuditResultDetailResponseBodyMediaAuditResultDetail,
            requestId: 'string',
        };
    }
    validate() {
        if (this.mediaAuditResultDetail && typeof this.mediaAuditResultDetail.validate === 'function') {
            this.mediaAuditResultDetail.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultDetailResponseBody = GetMediaAuditResultDetailResponseBody;
