// @alicloud/vod20170321/dist/models/GetMediaAuditResultResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetMediaAuditResultResponseBody = exports.GetMediaAuditResultResponseBodyMediaAuditResult = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResult = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResult = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResultTopList = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResultCounterList = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResult = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResultTopList = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResultCounterList = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResult = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResultTopList = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResultCounterList = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResult = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResultTopList = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResultCounterList = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResult = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResultTopList = exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResultCounterList = exports.GetMediaAuditResultResponseBodyMediaAuditResultTextResult = exports.GetMediaAuditResultResponseBodyMediaAuditResultImageResult = exports.GetMediaAuditResultResponseBodyMediaAuditResultImageResultResult = exports.GetMediaAuditResultResponseBodyMediaAuditResultAudioResult = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetMediaAuditResultResponseBodyMediaAuditResultAudioResult extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            scene: 'Scene',
            score: 'Score',
            suggestion: 'Suggestion',
        };
    }
    static types() {
        return {
            label: 'string',
            scene: 'string',
            score: 'string',
            suggestion: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultAudioResult = GetMediaAuditResultResponseBodyMediaAuditResultAudioResult;
class GetMediaAuditResultResponseBodyMediaAuditResultImageResultResult extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            scene: 'Scene',
            score: 'Score',
            suggestion: 'Suggestion',
        };
    }
    static types() {
        return {
            label: 'string',
            scene: 'string',
            score: 'string',
            suggestion: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultImageResultResult = GetMediaAuditResultResponseBodyMediaAuditResultImageResultResult;
class GetMediaAuditResultResponseBodyMediaAuditResultImageResult extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            result: 'Result',
            suggestion: 'Suggestion',
            type: 'Type',
            url: 'Url',
        };
    }
    static types() {
        return {
            label: 'string',
            result: { 'type': 'array', 'itemType': GetMediaAuditResultResponseBodyMediaAuditResultImageResultResult },
            suggestion: 'string',
            type: 'string',
            url: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.result)) {
            $dara.Model.validateArray(this.result);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultImageResult = GetMediaAuditResultResponseBodyMediaAuditResultImageResult;
class GetMediaAuditResultResponseBodyMediaAuditResultTextResult extends $dara.Model {
    static names() {
        return {
            content: 'Content',
            label: 'Label',
            scene: 'Scene',
            score: 'Score',
            suggestion: 'Suggestion',
            type: 'Type',
        };
    }
    static types() {
        return {
            content: 'string',
            label: 'string',
            scene: 'string',
            score: 'string',
            suggestion: 'string',
            type: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultTextResult = GetMediaAuditResultResponseBodyMediaAuditResultTextResult;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResultCounterList extends $dara.Model {
    static names() {
        return {
            count: 'Count',
            label: 'Label',
        };
    }
    static types() {
        return {
            count: 'number',
            label: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResultCounterList = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResultCounterList;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResultTopList extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
            url: 'Url',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
            url: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResultTopList = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResultTopList;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResult extends $dara.Model {
    static names() {
        return {
            averageScore: 'AverageScore',
            counterList: 'CounterList',
            label: 'Label',
            maxScore: 'MaxScore',
            suggestion: 'Suggestion',
            topList: 'TopList',
        };
    }
    static types() {
        return {
            averageScore: 'string',
            counterList: { 'type': 'array', 'itemType': GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResultCounterList },
            label: 'string',
            maxScore: 'string',
            suggestion: 'string',
            topList: { 'type': 'array', 'itemType': GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResultTopList },
        };
    }
    validate() {
        if (Array.isArray(this.counterList)) {
            $dara.Model.validateArray(this.counterList);
        }
        if (Array.isArray(this.topList)) {
            $dara.Model.validateArray(this.topList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResult = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResult;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResultCounterList extends $dara.Model {
    static names() {
        return {
            count: 'Count',
            label: 'Label',
        };
    }
    static types() {
        return {
            count: 'number',
            label: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResultCounterList = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResultCounterList;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResultTopList extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
            url: 'Url',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
            url: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResultTopList = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResultTopList;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResult extends $dara.Model {
    static names() {
        return {
            averageScore: 'AverageScore',
            counterList: 'CounterList',
            label: 'Label',
            maxScore: 'MaxScore',
            suggestion: 'Suggestion',
            topList: 'TopList',
        };
    }
    static types() {
        return {
            averageScore: 'string',
            counterList: { 'type': 'array', 'itemType': GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResultCounterList },
            label: 'string',
            maxScore: 'string',
            suggestion: 'string',
            topList: { 'type': 'array', 'itemType': GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResultTopList },
        };
    }
    validate() {
        if (Array.isArray(this.counterList)) {
            $dara.Model.validateArray(this.counterList);
        }
        if (Array.isArray(this.topList)) {
            $dara.Model.validateArray(this.topList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResult = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResult;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResultCounterList extends $dara.Model {
    static names() {
        return {
            count: 'Count',
            label: 'Label',
        };
    }
    static types() {
        return {
            count: 'number',
            label: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResultCounterList = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResultCounterList;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResultTopList extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
            url: 'Url',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
            url: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResultTopList = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResultTopList;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResult extends $dara.Model {
    static names() {
        return {
            averageScore: 'AverageScore',
            counterList: 'CounterList',
            label: 'Label',
            maxScore: 'MaxScore',
            suggestion: 'Suggestion',
            topList: 'TopList',
        };
    }
    static types() {
        return {
            averageScore: 'string',
            counterList: { 'type': 'array', 'itemType': GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResultCounterList },
            label: 'string',
            maxScore: 'string',
            suggestion: 'string',
            topList: { 'type': 'array', 'itemType': GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResultTopList },
        };
    }
    validate() {
        if (Array.isArray(this.counterList)) {
            $dara.Model.validateArray(this.counterList);
        }
        if (Array.isArray(this.topList)) {
            $dara.Model.validateArray(this.topList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResult = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResult;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResultCounterList extends $dara.Model {
    static names() {
        return {
            count: 'Count',
            label: 'Label',
        };
    }
    static types() {
        return {
            count: 'number',
            label: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResultCounterList = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResultCounterList;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResultTopList extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
            url: 'Url',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
            url: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResultTopList = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResultTopList;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResult extends $dara.Model {
    static names() {
        return {
            averageScore: 'AverageScore',
            counterList: 'CounterList',
            label: 'Label',
            maxScore: 'MaxScore',
            suggestion: 'Suggestion',
            topList: 'TopList',
        };
    }
    static types() {
        return {
            averageScore: 'string',
            counterList: { 'type': 'array', 'itemType': GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResultCounterList },
            label: 'string',
            maxScore: 'string',
            suggestion: 'string',
            topList: { 'type': 'array', 'itemType': GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResultTopList },
        };
    }
    validate() {
        if (Array.isArray(this.counterList)) {
            $dara.Model.validateArray(this.counterList);
        }
        if (Array.isArray(this.topList)) {
            $dara.Model.validateArray(this.topList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResult = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResult;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResultCounterList extends $dara.Model {
    static names() {
        return {
            count: 'Count',
            label: 'Label',
        };
    }
    static types() {
        return {
            count: 'number',
            label: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResultCounterList = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResultCounterList;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResultTopList extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
            url: 'Url',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
            url: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResultTopList = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResultTopList;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResult extends $dara.Model {
    static names() {
        return {
            averageScore: 'AverageScore',
            counterList: 'CounterList',
            label: 'Label',
            maxScore: 'MaxScore',
            suggestion: 'Suggestion',
            topList: 'TopList',
        };
    }
    static types() {
        return {
            averageScore: 'string',
            counterList: { 'type': 'array', 'itemType': GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResultCounterList },
            label: 'string',
            maxScore: 'string',
            suggestion: 'string',
            topList: { 'type': 'array', 'itemType': GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResultTopList },
        };
    }
    validate() {
        if (Array.isArray(this.counterList)) {
            $dara.Model.validateArray(this.counterList);
        }
        if (Array.isArray(this.topList)) {
            $dara.Model.validateArray(this.topList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResult = GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResult;
class GetMediaAuditResultResponseBodyMediaAuditResultVideoResult extends $dara.Model {
    static names() {
        return {
            adResult: 'AdResult',
            label: 'Label',
            liveResult: 'LiveResult',
            logoResult: 'LogoResult',
            pornResult: 'PornResult',
            suggestion: 'Suggestion',
            terrorismResult: 'TerrorismResult',
        };
    }
    static types() {
        return {
            adResult: GetMediaAuditResultResponseBodyMediaAuditResultVideoResultAdResult,
            label: 'string',
            liveResult: GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLiveResult,
            logoResult: GetMediaAuditResultResponseBodyMediaAuditResultVideoResultLogoResult,
            pornResult: GetMediaAuditResultResponseBodyMediaAuditResultVideoResultPornResult,
            suggestion: 'string',
            terrorismResult: GetMediaAuditResultResponseBodyMediaAuditResultVideoResultTerrorismResult,
        };
    }
    validate() {
        if (this.adResult && typeof this.adResult.validate === 'function') {
            this.adResult.validate();
        }
        if (this.liveResult && typeof this.liveResult.validate === 'function') {
            this.liveResult.validate();
        }
        if (this.logoResult && typeof this.logoResult.validate === 'function') {
            this.logoResult.validate();
        }
        if (this.pornResult && typeof this.pornResult.validate === 'function') {
            this.pornResult.validate();
        }
        if (this.terrorismResult && typeof this.terrorismResult.validate === 'function') {
            this.terrorismResult.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResultVideoResult = GetMediaAuditResultResponseBodyMediaAuditResultVideoResult;
class GetMediaAuditResultResponseBodyMediaAuditResult extends $dara.Model {
    static names() {
        return {
            abnormalModules: 'AbnormalModules',
            audioResult: 'AudioResult',
            imageResult: 'ImageResult',
            label: 'Label',
            suggestion: 'Suggestion',
            textResult: 'TextResult',
            videoResult: 'VideoResult',
        };
    }
    static types() {
        return {
            abnormalModules: 'string',
            audioResult: { 'type': 'array', 'itemType': GetMediaAuditResultResponseBodyMediaAuditResultAudioResult },
            imageResult: { 'type': 'array', 'itemType': GetMediaAuditResultResponseBodyMediaAuditResultImageResult },
            label: 'string',
            suggestion: 'string',
            textResult: { 'type': 'array', 'itemType': GetMediaAuditResultResponseBodyMediaAuditResultTextResult },
            videoResult: GetMediaAuditResultResponseBodyMediaAuditResultVideoResult,
        };
    }
    validate() {
        if (Array.isArray(this.audioResult)) {
            $dara.Model.validateArray(this.audioResult);
        }
        if (Array.isArray(this.imageResult)) {
            $dara.Model.validateArray(this.imageResult);
        }
        if (Array.isArray(this.textResult)) {
            $dara.Model.validateArray(this.textResult);
        }
        if (this.videoResult && typeof this.videoResult.validate === 'function') {
            this.videoResult.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBodyMediaAuditResult = GetMediaAuditResultResponseBodyMediaAuditResult;
class GetMediaAuditResultResponseBody extends $dara.Model {
    static names() {
        return {
            mediaAuditResult: 'MediaAuditResult',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            mediaAuditResult: GetMediaAuditResultResponseBodyMediaAuditResult,
            requestId: 'string',
        };
    }
    validate() {
        if (this.mediaAuditResult && typeof this.mediaAuditResult.validate === 'function') {
            this.mediaAuditResult.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultResponseBody = GetMediaAuditResultResponseBody;
