// @alicloud/vod20170321/dist/models/GetMezzanineInfoResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetMezzanineInfoResponseBody = exports.GetMezzanineInfoResponseBodyMezzanine = exports.GetMezzanineInfoResponseBodyMezzanineVideoStreamList = exports.GetMezzanineInfoResponseBodyMezzanineAudioStreamList = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetMezzanineInfoResponseBodyMezzanineAudioStreamList extends $dara.Model {
    static names() {
        return {
            bitrate: 'Bitrate',
            channelLayout: 'ChannelLayout',
            channels: 'Channels',
            codecLongName: 'CodecLongName',
            codecName: 'CodecName',
            codecTag: 'CodecTag',
            codecTagString: 'CodecTagString',
            codecTimeBase: 'CodecTimeBase',
            duration: 'Duration',
            index: 'Index',
            lang: 'Lang',
            numFrames: 'NumFrames',
            sampleFmt: 'SampleFmt',
            sampleRate: 'SampleRate',
            startTime: 'StartTime',
            timebase: 'Timebase',
        };
    }
    static types() {
        return {
            bitrate: 'string',
            channelLayout: 'string',
            channels: 'string',
            codecLongName: 'string',
            codecName: 'string',
            codecTag: 'string',
            codecTagString: 'string',
            codecTimeBase: 'string',
            duration: 'string',
            index: 'string',
            lang: 'string',
            numFrames: 'string',
            sampleFmt: 'string',
            sampleRate: 'string',
            startTime: 'string',
            timebase: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMezzanineInfoResponseBodyMezzanineAudioStreamList = GetMezzanineInfoResponseBodyMezzanineAudioStreamList;
class GetMezzanineInfoResponseBodyMezzanineVideoStreamList extends $dara.Model {
    static names() {
        return {
            avgFPS: 'AvgFPS',
            bitrate: 'Bitrate',
            codecLongName: 'CodecLongName',
            codecName: 'CodecName',
            codecTag: 'CodecTag',
            codecTagString: 'CodecTagString',
            codecTimeBase: 'CodecTimeBase',
            dar: 'Dar',
            duration: 'Duration',
            fps: 'Fps',
            HDRType: 'HDRType',
            hasBFrames: 'HasBFrames',
            height: 'Height',
            index: 'Index',
            lang: 'Lang',
            level: 'Level',
            numFrames: 'NumFrames',
            pixFmt: 'PixFmt',
            profile: 'Profile',
            rotate: 'Rotate',
            sar: 'Sar',
            startTime: 'StartTime',
            timebase: 'Timebase',
            width: 'Width',
        };
    }
    static types() {
        return {
            avgFPS: 'string',
            bitrate: 'string',
            codecLongName: 'string',
            codecName: 'string',
            codecTag: 'string',
            codecTagString: 'string',
            codecTimeBase: 'string',
            dar: 'string',
            duration: 'string',
            fps: 'string',
            HDRType: 'string',
            hasBFrames: 'string',
            height: 'string',
            index: 'string',
            lang: 'string',
            level: 'string',
            numFrames: 'string',
            pixFmt: 'string',
            profile: 'string',
            rotate: 'string',
            sar: 'string',
            startTime: 'string',
            timebase: 'string',
            width: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMezzanineInfoResponseBodyMezzanineVideoStreamList = GetMezzanineInfoResponseBodyMezzanineVideoStreamList;
class GetMezzanineInfoResponseBodyMezzanine extends $dara.Model {
    static names() {
        return {
            audioStreamList: 'AudioStreamList',
            bitrate: 'Bitrate',
            creationTime: 'CreationTime',
            duration: 'Duration',
            fileMD5: 'FileMD5',
            fileName: 'FileName',
            fileURL: 'FileURL',
            fps: 'Fps',
            height: 'Height',
            outputType: 'OutputType',
            preprocessStatus: 'PreprocessStatus',
            restoreExpiration: 'RestoreExpiration',
            restoreStatus: 'RestoreStatus',
            size: 'Size',
            status: 'Status',
            storageClass: 'StorageClass',
            videoId: 'VideoId',
            videoStreamList: 'VideoStreamList',
            width: 'Width',
        };
    }
    static types() {
        return {
            audioStreamList: { 'type': 'array', 'itemType': GetMezzanineInfoResponseBodyMezzanineAudioStreamList },
            bitrate: 'string',
            creationTime: 'string',
            duration: 'string',
            fileMD5: 'string',
            fileName: 'string',
            fileURL: 'string',
            fps: 'string',
            height: 'number',
            outputType: 'string',
            preprocessStatus: 'string',
            restoreExpiration: 'string',
            restoreStatus: 'string',
            size: 'number',
            status: 'string',
            storageClass: 'string',
            videoId: 'string',
            videoStreamList: { 'type': 'array', 'itemType': GetMezzanineInfoResponseBodyMezzanineVideoStreamList },
            width: 'number',
        };
    }
    validate() {
        if (Array.isArray(this.audioStreamList)) {
            $dara.Model.validateArray(this.audioStreamList);
        }
        if (Array.isArray(this.videoStreamList)) {
            $dara.Model.validateArray(this.videoStreamList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMezzanineInfoResponseBodyMezzanine = GetMezzanineInfoResponseBodyMezzanine;
class GetMezzanineInfoResponseBody extends $dara.Model {
    static names() {
        return {
            mezzanine: 'Mezzanine',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            mezzanine: GetMezzanineInfoResponseBodyMezzanine,
            requestId: 'string',
        };
    }
    validate() {
        if (this.mezzanine && typeof this.mezzanine.validate === 'function') {
            this.mezzanine.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMezzanineInfoResponseBody = GetMezzanineInfoResponseBody;
