// @alicloud/vod20170321/dist/models/GetTranscodeTemplateGroupResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetTranscodeTemplateGroupResponseBody = exports.GetTranscodeTemplateGroupResponseBodyTranscodeTemplateGroup = exports.GetTranscodeTemplateGroupResponseBodyTranscodeTemplateGroupTranscodeTemplateList = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetTranscodeTemplateGroupResponseBodyTranscodeTemplateGroupTranscodeTemplateList extends $dara.Model {
    static names() {
        return {
            audio: 'Audio',
            clip: 'Clip',
            container: 'Container',
            copyrightMark: 'CopyrightMark',
            definition: 'Definition',
            encryptSetting: 'EncryptSetting',
            muxConfig: 'MuxConfig',
            packageSetting: 'PackageSetting',
            rotate: 'Rotate',
            subtitleList: 'SubtitleList',
            templateName: 'TemplateName',
            traceMark: 'TraceMark',
            transConfig: 'TransConfig',
            transcodeFileRegular: 'TranscodeFileRegular',
            transcodeTemplateId: 'TranscodeTemplateId',
            type: 'Type',
            video: 'Video',
            watermarkIds: 'WatermarkIds',
        };
    }
    static types() {
        return {
            audio: 'string',
            clip: 'string',
            container: 'string',
            copyrightMark: 'string',
            definition: 'string',
            encryptSetting: 'string',
            muxConfig: 'string',
            packageSetting: 'string',
            rotate: 'string',
            subtitleList: 'string',
            templateName: 'string',
            traceMark: 'string',
            transConfig: 'string',
            transcodeFileRegular: 'string',
            transcodeTemplateId: 'string',
            type: 'string',
            video: 'string',
            watermarkIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
    validate() {
        if (Array.isArray(this.watermarkIds)) {
            $dara.Model.validateArray(this.watermarkIds);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscodeTemplateGroupResponseBodyTranscodeTemplateGroupTranscodeTemplateList = GetTranscodeTemplateGroupResponseBodyTranscodeTemplateGroupTranscodeTemplateList;
class GetTranscodeTemplateGroupResponseBodyTranscodeTemplateGroup extends $dara.Model {
    static names() {
        return {
            appId: 'AppId',
            creationTime: 'CreationTime',
            isDefault: 'IsDefault',
            locked: 'Locked',
            modifyTime: 'ModifyTime',
            name: 'Name',
            transcodeTemplateGroupId: 'TranscodeTemplateGroupId',
            transcodeTemplateList: 'TranscodeTemplateList',
        };
    }
    static types() {
        return {
            appId: 'string',
            creationTime: 'string',
            isDefault: 'string',
            locked: 'string',
            modifyTime: 'string',
            name: 'string',
            transcodeTemplateGroupId: 'string',
            transcodeTemplateList: { 'type': 'array', 'itemType': GetTranscodeTemplateGroupResponseBodyTranscodeTemplateGroupTranscodeTemplateList },
        };
    }
    validate() {
        if (Array.isArray(this.transcodeTemplateList)) {
            $dara.Model.validateArray(this.transcodeTemplateList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscodeTemplateGroupResponseBodyTranscodeTemplateGroup = GetTranscodeTemplateGroupResponseBodyTranscodeTemplateGroup;
class GetTranscodeTemplateGroupResponseBody extends $dara.Model {
    static names() {
        return {
            requestId: 'RequestId',
            transcodeTemplateGroup: 'TranscodeTemplateGroup',
        };
    }
    static types() {
        return {
            requestId: 'string',
            transcodeTemplateGroup: GetTranscodeTemplateGroupResponseBodyTranscodeTemplateGroup,
        };
    }
    validate() {
        if (this.transcodeTemplateGroup && typeof this.transcodeTemplateGroup.validate === 'function') {
            this.transcodeTemplateGroup.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscodeTemplateGroupResponseBody = GetTranscodeTemplateGroupResponseBody;
