// @alicloud/vod20170321/dist/models/GetTranscodeTaskResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetTranscodeTaskResponseBody = exports.GetTranscodeTaskResponseBodyTranscodeTask = exports.GetTranscodeTaskResponseBodyTranscodeTaskTranscodeJobInfoList = exports.GetTranscodeTaskResponseBodyTranscodeTaskTranscodeJobInfoListOutputFile = exports.GetTranscodeTaskResponseBodyTranscodeJobInfoList = exports.GetTranscodeTaskResponseBodyTranscodeJobInfoListOutputFile = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetTranscodeTaskResponseBodyTranscodeJobInfoListOutputFile extends $dara.Model {
    static names() {
        return {
            audioStreamList: 'AudioStreamList',
            bitrate: 'Bitrate',
            duration: 'Duration',
            encryption: 'Encryption',
            filesize: 'Filesize',
            format: 'Format',
            fps: 'Fps',
            height: 'Height',
            outputFileUrl: 'OutputFileUrl',
            subtitleStreamList: 'SubtitleStreamList',
            videoStreamList: 'VideoStreamList',
            watermarkIdList: 'WatermarkIdList',
            width: 'Width',
        };
    }
    static types() {
        return {
            audioStreamList: 'string',
            bitrate: 'string',
            duration: 'string',
            encryption: 'string',
            filesize: 'number',
            format: 'string',
            fps: 'string',
            height: 'string',
            outputFileUrl: 'string',
            subtitleStreamList: 'string',
            videoStreamList: 'string',
            watermarkIdList: { 'type': 'array', 'itemType': 'string' },
            width: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.watermarkIdList)) {
            $dara.Model.validateArray(this.watermarkIdList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscodeTaskResponseBodyTranscodeJobInfoListOutputFile = GetTranscodeTaskResponseBodyTranscodeJobInfoListOutputFile;
class GetTranscodeTaskResponseBodyTranscodeJobInfoList extends $dara.Model {
    static names() {
        return {
            completeTime: 'CompleteTime',
            creationTime: 'CreationTime',
            definition: 'Definition',
            errorCode: 'ErrorCode',
            errorMessage: 'ErrorMessage',
            inputFileUrl: 'InputFileUrl',
            outputFile: 'OutputFile',
            priority: 'Priority',
            transcodeJobId: 'TranscodeJobId',
            transcodeJobStatus: 'TranscodeJobStatus',
            transcodeProgress: 'TranscodeProgress',
            transcodeTemplateId: 'TranscodeTemplateId',
        };
    }
    static types() {
        return {
            completeTime: 'string',
            creationTime: 'string',
            definition: 'string',
            errorCode: 'string',
            errorMessage: 'string',
            inputFileUrl: 'string',
            outputFile: GetTranscodeTaskResponseBodyTranscodeJobInfoListOutputFile,
            priority: 'string',
            transcodeJobId: 'string',
            transcodeJobStatus: 'string',
            transcodeProgress: 'number',
            transcodeTemplateId: 'string',
        };
    }
    validate() {
        if (this.outputFile && typeof this.outputFile.validate === 'function') {
            this.outputFile.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscodeTaskResponseBodyTranscodeJobInfoList = GetTranscodeTaskResponseBodyTranscodeJobInfoList;
class GetTranscodeTaskResponseBodyTranscodeTaskTranscodeJobInfoListOutputFile extends $dara.Model {
    static names() {
        return {
            audioStreamList: 'AudioStreamList',
            bitrate: 'Bitrate',
            duration: 'Duration',
            encryption: 'Encryption',
            filesize: 'Filesize',
            format: 'Format',
            fps: 'Fps',
            height: 'Height',
            outputFileUrl: 'OutputFileUrl',
            subtitleStreamList: 'SubtitleStreamList',
            videoStreamList: 'VideoStreamList',
            watermarkIdList: 'WatermarkIdList',
            width: 'Width',
        };
    }
    static types() {
        return {
            audioStreamList: 'string',
            bitrate: 'string',
            duration: 'string',
            encryption: 'string',
            filesize: 'number',
            format: 'string',
            fps: 'string',
            height: 'string',
            outputFileUrl: 'string',
            subtitleStreamList: 'string',
            videoStreamList: 'string',
            watermarkIdList: { 'type': 'array', 'itemType': 'string' },
            width: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.watermarkIdList)) {
            $dara.Model.validateArray(this.watermarkIdList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscodeTaskResponseBodyTranscodeTaskTranscodeJobInfoListOutputFile = GetTranscodeTaskResponseBodyTranscodeTaskTranscodeJobInfoListOutputFile;
class GetTranscodeTaskResponseBodyTranscodeTaskTranscodeJobInfoList extends $dara.Model {
    static names() {
        return {
            completeTime: 'CompleteTime',
            creationTime: 'CreationTime',
            definition: 'Definition',
            errorCode: 'ErrorCode',
            errorMessage: 'ErrorMessage',
            inputFileUrl: 'InputFileUrl',
            outputFile: 'OutputFile',
            priority: 'Priority',
            transcodeJobId: 'TranscodeJobId',
            transcodeJobStatus: 'TranscodeJobStatus',
            transcodeProgress: 'TranscodeProgress',
            transcodeTemplateId: 'TranscodeTemplateId',
        };
    }
    static types() {
        return {
            completeTime: 'string',
            creationTime: 'string',
            definition: 'string',
            errorCode: 'string',
            errorMessage: 'string',
            inputFileUrl: 'string',
            outputFile: GetTranscodeTaskResponseBodyTranscodeTaskTranscodeJobInfoListOutputFile,
            priority: 'string',
            transcodeJobId: 'string',
            transcodeJobStatus: 'string',
            transcodeProgress: 'number',
            transcodeTemplateId: 'string',
        };
    }
    validate() {
        if (this.outputFile && typeof this.outputFile.validate === 'function') {
            this.outputFile.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscodeTaskResponseBodyTranscodeTaskTranscodeJobInfoList = GetTranscodeTaskResponseBodyTranscodeTaskTranscodeJobInfoList;
class GetTranscodeTaskResponseBodyTranscodeTask extends $dara.Model {
    static names() {
        return {
            completeTime: 'CompleteTime',
            creationTime: 'CreationTime',
            taskStatus: 'TaskStatus',
            transcodeJobInfoList: 'TranscodeJobInfoList',
            transcodeTaskId: 'TranscodeTaskId',
            transcodeTemplateGroupId: 'TranscodeTemplateGroupId',
            trigger: 'Trigger',
            videoId: 'VideoId',
        };
    }
    static types() {
        return {
            completeTime: 'string',
            creationTime: 'string',
            taskStatus: 'string',
            transcodeJobInfoList: { 'type': 'array', 'itemType': GetTranscodeTaskResponseBodyTranscodeTaskTranscodeJobInfoList },
            transcodeTaskId: 'string',
            transcodeTemplateGroupId: 'string',
            trigger: 'string',
            videoId: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.transcodeJobInfoList)) {
            $dara.Model.validateArray(this.transcodeJobInfoList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscodeTaskResponseBodyTranscodeTask = GetTranscodeTaskResponseBodyTranscodeTask;
class GetTranscodeTaskResponseBody extends $dara.Model {
    static names() {
        return {
            nonExistJobIds: 'NonExistJobIds',
            requestId: 'RequestId',
            transcodeJobInfoList: 'TranscodeJobInfoList',
            transcodeTask: 'TranscodeTask',
        };
    }
    static types() {
        return {
            nonExistJobIds: { 'type': 'array', 'itemType': 'string' },
            requestId: 'string',
            transcodeJobInfoList: { 'type': 'array', 'itemType': GetTranscodeTaskResponseBodyTranscodeJobInfoList },
            transcodeTask: GetTranscodeTaskResponseBodyTranscodeTask,
        };
    }
    validate() {
        if (Array.isArray(this.nonExistJobIds)) {
            $dara.Model.validateArray(this.nonExistJobIds);
        }
        if (Array.isArray(this.transcodeJobInfoList)) {
            $dara.Model.validateArray(this.transcodeJobInfoList);
        }
        if (this.transcodeTask && typeof this.transcodeTask.validate === 'function') {
            this.transcodeTask.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscodeTaskResponseBody = GetTranscodeTaskResponseBody;
