// @alicloud/vod20170321/dist/models/ListLiveRecordVideoResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListLiveRecordVideoResponseBody = exports.ListLiveRecordVideoResponseBodyLiveRecordVideoList = exports.ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideo = exports.ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideoVideo = exports.ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideoVideoSnapshots = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideoVideoSnapshots extends $dara.Model {
    static names() {
        return {
            snapshot: 'Snapshot',
        };
    }
    static types() {
        return {
            snapshot: { 'type': 'array', 'itemType': 'string' },
        };
    }
    validate() {
        if (Array.isArray(this.snapshot)) {
            $dara.Model.validateArray(this.snapshot);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideoVideoSnapshots = ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideoVideoSnapshots;
class ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideoVideo extends $dara.Model {
    static names() {
        return {
            cateId: 'CateId',
            cateName: 'CateName',
            coverURL: 'CoverURL',
            creationTime: 'CreationTime',
            description: 'Description',
            duration: 'Duration',
            modifyTime: 'ModifyTime',
            size: 'Size',
            snapshots: 'Snapshots',
            status: 'Status',
            tags: 'Tags',
            templateGroupId: 'TemplateGroupId',
            title: 'Title',
            videoId: 'VideoId',
        };
    }
    static types() {
        return {
            cateId: 'number',
            cateName: 'string',
            coverURL: 'string',
            creationTime: 'string',
            description: 'string',
            duration: 'number',
            modifyTime: 'string',
            size: 'number',
            snapshots: ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideoVideoSnapshots,
            status: 'string',
            tags: 'string',
            templateGroupId: 'string',
            title: 'string',
            videoId: 'string',
        };
    }
    validate() {
        if (this.snapshots && typeof this.snapshots.validate === 'function') {
            this.snapshots.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideoVideo = ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideoVideo;
class ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideo extends $dara.Model {
    static names() {
        return {
            appName: 'AppName',
            domainName: 'DomainName',
            playlistId: 'PlaylistId',
            recordEndTime: 'RecordEndTime',
            recordStartTime: 'RecordStartTime',
            streamName: 'StreamName',
            video: 'Video',
        };
    }
    static types() {
        return {
            appName: 'string',
            domainName: 'string',
            playlistId: 'string',
            recordEndTime: 'string',
            recordStartTime: 'string',
            streamName: 'string',
            video: ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideoVideo,
        };
    }
    validate() {
        if (this.video && typeof this.video.validate === 'function') {
            this.video.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideo = ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideo;
class ListLiveRecordVideoResponseBodyLiveRecordVideoList extends $dara.Model {
    static names() {
        return {
            liveRecordVideo: 'LiveRecordVideo',
        };
    }
    static types() {
        return {
            liveRecordVideo: { 'type': 'array', 'itemType': ListLiveRecordVideoResponseBodyLiveRecordVideoListLiveRecordVideo },
        };
    }
    validate() {
        if (Array.isArray(this.liveRecordVideo)) {
            $dara.Model.validateArray(this.liveRecordVideo);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.ListLiveRecordVideoResponseBodyLiveRecordVideoList = ListLiveRecordVideoResponseBodyLiveRecordVideoList;
class ListLiveRecordVideoResponseBody extends $dara.Model {
    static names() {
        return {
            liveRecordVideoList: 'LiveRecordVideoList',
            requestId: 'RequestId',
            total: 'Total',
        };
    }
    static types() {
        return {
            liveRecordVideoList: ListLiveRecordVideoResponseBodyLiveRecordVideoList,
            requestId: 'string',
            total: 'number',
        };
    }
    validate() {
        if (this.liveRecordVideoList && typeof this.liveRecordVideoList.validate === 'function') {
            this.liveRecordVideoList.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.ListLiveRecordVideoResponseBody = ListLiveRecordVideoResponseBody;
