// @alicloud/vod20170321/dist/models/SearchEditingProjectResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchEditingProjectResponseBody = exports.SearchEditingProjectResponseBodyProjectList = exports.SearchEditingProjectResponseBodyProjectListProject = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class SearchEditingProjectResponseBodyProjectListProject extends $dara.Model {
    static names() {
        return {
            coverURL: 'CoverURL',
            creationTime: 'CreationTime',
            description: 'Description',
            duration: 'Duration',
            modifiedTime: 'ModifiedTime',
            projectId: 'ProjectId',
            regionId: 'RegionId',
            status: 'Status',
            storageLocation: 'StorageLocation',
            title: 'Title',
        };
    }
    static types() {
        return {
            coverURL: 'string',
            creationTime: 'string',
            description: 'string',
            duration: 'number',
            modifiedTime: 'string',
            projectId: 'string',
            regionId: 'string',
            status: 'string',
            storageLocation: 'string',
            title: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchEditingProjectResponseBodyProjectListProject = SearchEditingProjectResponseBodyProjectListProject;
class SearchEditingProjectResponseBodyProjectList extends $dara.Model {
    static names() {
        return {
            project: 'Project',
        };
    }
    static types() {
        return {
            project: { 'type': 'array', 'itemType': SearchEditingProjectResponseBodyProjectListProject },
        };
    }
    validate() {
        if (Array.isArray(this.project)) {
            $dara.Model.validateArray(this.project);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchEditingProjectResponseBodyProjectList = SearchEditingProjectResponseBodyProjectList;
class SearchEditingProjectResponseBody extends $dara.Model {
    static names() {
        return {
            projectList: 'ProjectList',
            requestId: 'RequestId',
            total: 'Total',
        };
    }
    static types() {
        return {
            projectList: SearchEditingProjectResponseBodyProjectList,
            requestId: 'string',
            total: 'number',
        };
    }
    validate() {
        if (this.projectList && typeof this.projectList.validate === 'function') {
            this.projectList.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchEditingProjectResponseBody = SearchEditingProjectResponseBody;
