// @alicloud/vod20170321/dist/models/SearchMediaResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchMediaResponseBody = exports.SearchMediaResponseBodyMediaList = exports.SearchMediaResponseBodyMediaListVideo = exports.SearchMediaResponseBodyMediaListImage = exports.SearchMediaResponseBodyMediaListAudio = exports.SearchMediaResponseBodyMediaListAttachedMedia = exports.SearchMediaResponseBodyMediaListAttachedMediaCategories = exports.SearchMediaResponseBodyMediaListAiRoughData = exports.SearchMediaResponseBodyMediaListAiData = exports.SearchMediaResponseBodyMediaListAiDataOcrInfo = exports.SearchMediaResponseBodyMediaListAiDataAiLabelInfo = exports.SearchMediaResponseBodyMediaListAiDataAiLabelInfoOccurrences = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class SearchMediaResponseBodyMediaListAiDataAiLabelInfoOccurrences extends $dara.Model {
    static names() {
        return {
            from: 'From',
            score: 'Score',
            to: 'To',
        };
    }
    static types() {
        return {
            from: 'number',
            score: 'number',
            to: 'number',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchMediaResponseBodyMediaListAiDataAiLabelInfoOccurrences = SearchMediaResponseBodyMediaListAiDataAiLabelInfoOccurrences;
class SearchMediaResponseBodyMediaListAiDataAiLabelInfo extends $dara.Model {
    static names() {
        return {
            category: 'Category',
            labelId: 'LabelId',
            labelName: 'LabelName',
            occurrences: 'Occurrences',
        };
    }
    static types() {
        return {
            category: 'string',
            labelId: 'string',
            labelName: 'string',
            occurrences: { 'type': 'array', 'itemType': SearchMediaResponseBodyMediaListAiDataAiLabelInfoOccurrences },
        };
    }
    validate() {
        if (Array.isArray(this.occurrences)) {
            $dara.Model.validateArray(this.occurrences);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchMediaResponseBodyMediaListAiDataAiLabelInfo = SearchMediaResponseBodyMediaListAiDataAiLabelInfo;
class SearchMediaResponseBodyMediaListAiDataOcrInfo extends $dara.Model {
    static names() {
        return {
            content: 'Content',
            from: 'From',
            to: 'To',
        };
    }
    static types() {
        return {
            content: 'string',
            from: 'number',
            to: 'number',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchMediaResponseBodyMediaListAiDataOcrInfo = SearchMediaResponseBodyMediaListAiDataOcrInfo;
class SearchMediaResponseBodyMediaListAiData extends $dara.Model {
    static names() {
        return {
            aiLabelInfo: 'AiLabelInfo',
            ocrInfo: 'OcrInfo',
        };
    }
    static types() {
        return {
            aiLabelInfo: { 'type': 'array', 'itemType': SearchMediaResponseBodyMediaListAiDataAiLabelInfo },
            ocrInfo: { 'type': 'array', 'itemType': SearchMediaResponseBodyMediaListAiDataOcrInfo },
        };
    }
    validate() {
        if (Array.isArray(this.aiLabelInfo)) {
            $dara.Model.validateArray(this.aiLabelInfo);
        }
        if (Array.isArray(this.ocrInfo)) {
            $dara.Model.validateArray(this.ocrInfo);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchMediaResponseBodyMediaListAiData = SearchMediaResponseBodyMediaListAiData;
class SearchMediaResponseBodyMediaListAiRoughData extends $dara.Model {
    static names() {
        return {
            aiCategory: 'AiCategory',
            aiJobId: 'AiJobId',
            saveType: 'SaveType',
            status: 'Status',
        };
    }
    static types() {
        return {
            aiCategory: 'string',
            aiJobId: 'string',
            saveType: 'string',
            status: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchMediaResponseBodyMediaListAiRoughData = SearchMediaResponseBodyMediaListAiRoughData;
class SearchMediaResponseBodyMediaListAttachedMediaCategories extends $dara.Model {
    static names() {
        return {
            cateId: 'CateId',
            cateName: 'CateName',
            level: 'Level',
            parentId: 'ParentId',
        };
    }
    static types() {
        return {
            cateId: 'number',
            cateName: 'string',
            level: 'number',
            parentId: 'number',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchMediaResponseBodyMediaListAttachedMediaCategories = SearchMediaResponseBodyMediaListAttachedMediaCategories;
class SearchMediaResponseBodyMediaListAttachedMedia extends $dara.Model {
    static names() {
        return {
            appId: 'AppId',
            businessType: 'BusinessType',
            categories: 'Categories',
            creationTime: 'CreationTime',
            description: 'Description',
            mediaId: 'MediaId',
            modificationTime: 'ModificationTime',
            status: 'Status',
            storageLocation: 'StorageLocation',
            tags: 'Tags',
            title: 'Title',
            URL: 'URL',
        };
    }
    static types() {
        return {
            appId: 'string',
            businessType: 'string',
            categories: { 'type': 'array', 'itemType': SearchMediaResponseBodyMediaListAttachedMediaCategories },
            creationTime: 'string',
            description: 'string',
            mediaId: 'string',
            modificationTime: 'string',
            status: 'string',
            storageLocation: 'string',
            tags: 'string',
            title: 'string',
            URL: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.categories)) {
            $dara.Model.validateArray(this.categories);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchMediaResponseBodyMediaListAttachedMedia = SearchMediaResponseBodyMediaListAttachedMedia;
class SearchMediaResponseBodyMediaListAudio extends $dara.Model {
    static names() {
        return {
            appId: 'AppId',
            audioId: 'AudioId',
            cateId: 'CateId',
            cateName: 'CateName',
            coverURL: 'CoverURL',
            creationTime: 'CreationTime',
            description: 'Description',
            downloadSwitch: 'DownloadSwitch',
            duration: 'Duration',
            mediaSource: 'MediaSource',
            modificationTime: 'ModificationTime',
            preprocessStatus: 'PreprocessStatus',
            referenceId: 'ReferenceId',
            restoreExpiration: 'RestoreExpiration',
            restoreStatus: 'RestoreStatus',
            size: 'Size',
            snapshots: 'Snapshots',
            spriteSnapshots: 'SpriteSnapshots',
            status: 'Status',
            storageClass: 'StorageClass',
            storageLocation: 'StorageLocation',
            tags: 'Tags',
            title: 'Title',
            transcodeMode: 'TranscodeMode',
        };
    }
    static types() {
        return {
            appId: 'string',
            audioId: 'string',
            cateId: 'number',
            cateName: 'string',
            coverURL: 'string',
            creationTime: 'string',
            description: 'string',
            downloadSwitch: 'string',
            duration: 'number',
            mediaSource: 'string',
            modificationTime: 'string',
            preprocessStatus: 'string',
            referenceId: 'string',
            restoreExpiration: 'string',
            restoreStatus: 'string',
            size: 'number',
            snapshots: { 'type': 'array', 'itemType': 'string' },
            spriteSnapshots: { 'type': 'array', 'itemType': 'string' },
            status: 'string',
            storageClass: 'string',
            storageLocation: 'string',
            tags: 'string',
            title: 'string',
            transcodeMode: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.snapshots)) {
            $dara.Model.validateArray(this.snapshots);
        }
        if (Array.isArray(this.spriteSnapshots)) {
            $dara.Model.validateArray(this.spriteSnapshots);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchMediaResponseBodyMediaListAudio = SearchMediaResponseBodyMediaListAudio;
class SearchMediaResponseBodyMediaListImage extends $dara.Model {
    static names() {
        return {
            appId: 'AppId',
            cateId: 'CateId',
            cateName: 'CateName',
            creationTime: 'CreationTime',
            description: 'Description',
            imageId: 'ImageId',
            modificationTime: 'ModificationTime',
            status: 'Status',
            storageLocation: 'StorageLocation',
            tags: 'Tags',
            title: 'Title',
            URL: 'URL',
        };
    }
    static types() {
        return {
            appId: 'string',
            cateId: 'number',
            cateName: 'string',
            creationTime: 'string',
            description: 'string',
            imageId: 'string',
            modificationTime: 'string',
            status: 'string',
            storageLocation: 'string',
            tags: 'string',
            title: 'string',
            URL: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchMediaResponseBodyMediaListImage = SearchMediaResponseBodyMediaListImage;
class SearchMediaResponseBodyMediaListVideo extends $dara.Model {
    static names() {
        return {
            appId: 'AppId',
            cateId: 'CateId',
            cateName: 'CateName',
            coverURL: 'CoverURL',
            creationTime: 'CreationTime',
            description: 'Description',
            downloadSwitch: 'DownloadSwitch',
            duration: 'Duration',
            mediaSource: 'MediaSource',
            modificationTime: 'ModificationTime',
            preprocessStatus: 'PreprocessStatus',
            referenceId: 'ReferenceId',
            restoreExpiration: 'RestoreExpiration',
            restoreStatus: 'RestoreStatus',
            size: 'Size',
            snapshots: 'Snapshots',
            spriteSnapshots: 'SpriteSnapshots',
            status: 'Status',
            storageClass: 'StorageClass',
            storageLocation: 'StorageLocation',
            tags: 'Tags',
            title: 'Title',
            transcodeMode: 'TranscodeMode',
            videoId: 'VideoId',
        };
    }
    static types() {
        return {
            appId: 'string',
            cateId: 'number',
            cateName: 'string',
            coverURL: 'string',
            creationTime: 'string',
            description: 'string',
            downloadSwitch: 'string',
            duration: 'number',
            mediaSource: 'string',
            modificationTime: 'string',
            preprocessStatus: 'string',
            referenceId: 'string',
            restoreExpiration: 'string',
            restoreStatus: 'string',
            size: 'number',
            snapshots: { 'type': 'array', 'itemType': 'string' },
            spriteSnapshots: { 'type': 'array', 'itemType': 'string' },
            status: 'string',
            storageClass: 'string',
            storageLocation: 'string',
            tags: 'string',
            title: 'string',
            transcodeMode: 'string',
            videoId: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.snapshots)) {
            $dara.Model.validateArray(this.snapshots);
        }
        if (Array.isArray(this.spriteSnapshots)) {
            $dara.Model.validateArray(this.spriteSnapshots);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchMediaResponseBodyMediaListVideo = SearchMediaResponseBodyMediaListVideo;
class SearchMediaResponseBodyMediaList extends $dara.Model {
    static names() {
        return {
            aiData: 'AiData',
            aiRoughData: 'AiRoughData',
            attachedMedia: 'AttachedMedia',
            audio: 'Audio',
            creationTime: 'CreationTime',
            image: 'Image',
            mediaId: 'MediaId',
            mediaType: 'MediaType',
            video: 'Video',
        };
    }
    static types() {
        return {
            aiData: SearchMediaResponseBodyMediaListAiData,
            aiRoughData: SearchMediaResponseBodyMediaListAiRoughData,
            attachedMedia: SearchMediaResponseBodyMediaListAttachedMedia,
            audio: SearchMediaResponseBodyMediaListAudio,
            creationTime: 'string',
            image: SearchMediaResponseBodyMediaListImage,
            mediaId: 'string',
            mediaType: 'string',
            video: SearchMediaResponseBodyMediaListVideo,
        };
    }
    validate() {
        if (this.aiData && typeof this.aiData.validate === 'function') {
            this.aiData.validate();
        }
        if (this.aiRoughData && typeof this.aiRoughData.validate === 'function') {
            this.aiRoughData.validate();
        }
        if (this.attachedMedia && typeof this.attachedMedia.validate === 'function') {
            this.attachedMedia.validate();
        }
        if (this.audio && typeof this.audio.validate === 'function') {
            this.audio.validate();
        }
        if (this.image && typeof this.image.validate === 'function') {
            this.image.validate();
        }
        if (this.video && typeof this.video.validate === 'function') {
            this.video.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchMediaResponseBodyMediaList = SearchMediaResponseBodyMediaList;
class SearchMediaResponseBody extends $dara.Model {
    static names() {
        return {
            mediaList: 'MediaList',
            requestId: 'RequestId',
            scrollToken: 'ScrollToken',
            total: 'Total',
        };
    }
    static types() {
        return {
            mediaList: { 'type': 'array', 'itemType': SearchMediaResponseBodyMediaList },
            requestId: 'string',
            scrollToken: 'string',
            total: 'number',
        };
    }
    validate() {
        if (Array.isArray(this.mediaList)) {
            $dara.Model.validateArray(this.mediaList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SearchMediaResponseBody = SearchMediaResponseBody;
