// @alicloud/vod20170321/dist/models/SubmitTranscodeJobsResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubmitTranscodeJobsResponseBody = exports.SubmitTranscodeJobsResponseBodyTranscodeJobs = exports.SubmitTranscodeJobsResponseBodyTranscodeJobsTranscodeJob = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class SubmitTranscodeJobsResponseBodyTranscodeJobsTranscodeJob extends $dara.Model {
    static names() {
        return {
            jobId: 'JobId',
        };
    }
    static types() {
        return {
            jobId: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SubmitTranscodeJobsResponseBodyTranscodeJobsTranscodeJob = SubmitTranscodeJobsResponseBodyTranscodeJobsTranscodeJob;
class SubmitTranscodeJobsResponseBodyTranscodeJobs extends $dara.Model {
    static names() {
        return {
            transcodeJob: 'TranscodeJob',
        };
    }
    static types() {
        return {
            transcodeJob: { 'type': 'array', 'itemType': SubmitTranscodeJobsResponseBodyTranscodeJobsTranscodeJob },
        };
    }
    validate() {
        if (Array.isArray(this.transcodeJob)) {
            $dara.Model.validateArray(this.transcodeJob);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SubmitTranscodeJobsResponseBodyTranscodeJobs = SubmitTranscodeJobsResponseBodyTranscodeJobs;
class SubmitTranscodeJobsResponseBody extends $dara.Model {
    static names() {
        return {
            requestId: 'RequestId',
            transcodeJobs: 'TranscodeJobs',
            transcodeTaskId: 'TranscodeTaskId',
        };
    }
    static types() {
        return {
            requestId: 'string',
            transcodeJobs: SubmitTranscodeJobsResponseBodyTranscodeJobs,
            transcodeTaskId: 'string',
        };
    }
    validate() {
        if (this.transcodeJobs && typeof this.transcodeJobs.validate === 'function') {
            this.transcodeJobs.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SubmitTranscodeJobsResponseBody = SubmitTranscodeJobsResponseBody;
