// @alicloud/vod20170321@3.11.0/dist/models/AppInfoDto.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppInfoDTO = exports.AppInfoDTOPlatforms = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
/**
 */
class AppInfoDTOPlatforms extends $dara.Model {
    static names() {
        return {
            itemId: 'ItemId',
            licenseItemIds: 'LicenseItemIds',
            pkgName: 'PkgName',
            pkgSignature: 'PkgSignature',
            platformType: 'PlatformType',
            type: 'Type',
        };
    }
    static types() {
        return {
            itemId: 'string',
            licenseItemIds: { 'type': 'array', 'itemType': 'string' },
            pkgName: 'string',
            pkgSignature: 'string',
            platformType: 'number',
            type: 'number',
        };
    }
    validate() {
        if (Array.isArray(this.licenseItemIds)) {
            $dara.Model.validateArray(this.licenseItemIds);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.AppInfoDTOPlatforms = AppInfoDTOPlatforms;
class AppInfoDTO extends $dara.Model {
    static names() {
        return {
            appName: 'AppName',
            appType: 'AppType',
            creationTime: 'CreationTime',
            gmtCreate: 'GmtCreate',
            gmtModified: 'GmtModified',
            itemId: 'ItemId',
            modificationTime: 'ModificationTime',
            platforms: 'Platforms',
            userId: 'UserId',
        };
    }
    static types() {
        return {
            appName: 'string',
            appType: 'number',
            creationTime: 'string',
            gmtCreate: 'string',
            gmtModified: 'string',
            itemId: 'string',
            modificationTime: 'string',
            platforms: { 'type': 'array', 'itemType': AppInfoDTOPlatforms },
            userId: 'number',
        };
    }
    validate() {
        if (Array.isArray(this.platforms)) {
            $dara.Model.validateArray(this.platforms);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.AppInfoDTO = AppInfoDTO;
