// @alicloud/vod20170321@3.11.0/dist/models/BatchGetMediaInfosResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchGetMediaInfosResponseBody = exports.BatchGetMediaInfosResponseBodyMediaInfos = exports.BatchGetMediaInfosResponseBodyMediaInfosPlayInfoList = exports.BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfo = exports.BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfoVideoStreamList = exports.BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfoAudioStreamList = exports.BatchGetMediaInfosResponseBodyMediaInfosMediaInfo = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class BatchGetMediaInfosResponseBodyMediaInfosMediaInfo extends $dara.Model {
    static names() {
        return {
            appId: 'AppId',
            cateId: 'CateId',
            cateName: 'CateName',
            coverURL: 'CoverURL',
            creationTime: 'CreationTime',
            description: 'Description',
            downloadSwitch: 'DownloadSwitch',
            mediaId: 'MediaId',
            modificationTime: 'ModificationTime',
            referenceId: 'ReferenceId',
            restoreExpiration: 'RestoreExpiration',
            restoreStatus: 'RestoreStatus',
            snapshots: 'Snapshots',
            status: 'Status',
            storageClass: 'StorageClass',
            storageLocation: 'StorageLocation',
            tags: 'Tags',
            templateGroupId: 'TemplateGroupId',
            title: 'Title',
            userData: 'UserData',
        };
    }
    static types() {
        return {
            appId: 'string',
            cateId: 'number',
            cateName: 'string',
            coverURL: 'string',
            creationTime: 'string',
            description: 'string',
            downloadSwitch: 'string',
            mediaId: 'string',
            modificationTime: 'string',
            referenceId: 'string',
            restoreExpiration: 'string',
            restoreStatus: 'string',
            snapshots: { 'type': 'array', 'itemType': 'string' },
            status: 'string',
            storageClass: 'string',
            storageLocation: 'string',
            tags: 'string',
            templateGroupId: 'string',
            title: 'string',
            userData: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.snapshots)) {
            $dara.Model.validateArray(this.snapshots);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.BatchGetMediaInfosResponseBodyMediaInfosMediaInfo = BatchGetMediaInfosResponseBodyMediaInfosMediaInfo;
class BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfoAudioStreamList extends $dara.Model {
    static names() {
        return {
            bitrate: 'Bitrate',
            channelLayout: 'ChannelLayout',
            channels: 'Channels',
            codecLongName: 'CodecLongName',
            codecName: 'CodecName',
            codecTag: 'CodecTag',
            codecTagString: 'CodecTagString',
            codecTimeBase: 'CodecTimeBase',
            duration: 'Duration',
            index: 'Index',
            lang: 'Lang',
            numFrames: 'NumFrames',
            sampleFmt: 'SampleFmt',
            sampleRate: 'SampleRate',
            startTime: 'StartTime',
            timebase: 'Timebase',
        };
    }
    static types() {
        return {
            bitrate: 'string',
            channelLayout: 'string',
            channels: 'string',
            codecLongName: 'string',
            codecName: 'string',
            codecTag: 'string',
            codecTagString: 'string',
            codecTimeBase: 'string',
            duration: 'string',
            index: 'string',
            lang: 'string',
            numFrames: 'string',
            sampleFmt: 'string',
            sampleRate: 'string',
            startTime: 'string',
            timebase: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfoAudioStreamList = BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfoAudioStreamList;
class BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfoVideoStreamList extends $dara.Model {
    static names() {
        return {
            avgFPS: 'AvgFPS',
            bitrate: 'Bitrate',
            codecLongName: 'CodecLongName',
            codecName: 'CodecName',
            codecTag: 'CodecTag',
            codecTagString: 'CodecTagString',
            codecTimeBase: 'CodecTimeBase',
            dar: 'Dar',
            duration: 'Duration',
            fps: 'Fps',
            HDRType: 'HDRType',
            hasBFrames: 'HasBFrames',
            height: 'Height',
            index: 'Index',
            lang: 'Lang',
            level: 'Level',
            numFrames: 'NumFrames',
            pixFmt: 'PixFmt',
            profile: 'Profile',
            rotate: 'Rotate',
            sar: 'Sar',
            startTime: 'StartTime',
            timebase: 'Timebase',
            width: 'Width',
        };
    }
    static types() {
        return {
            avgFPS: 'string',
            bitrate: 'string',
            codecLongName: 'string',
            codecName: 'string',
            codecTag: 'string',
            codecTagString: 'string',
            codecTimeBase: 'string',
            dar: 'string',
            duration: 'string',
            fps: 'string',
            HDRType: 'string',
            hasBFrames: 'string',
            height: 'string',
            index: 'string',
            lang: 'string',
            level: 'string',
            numFrames: 'string',
            pixFmt: 'string',
            profile: 'string',
            rotate: 'string',
            sar: 'string',
            startTime: 'string',
            timebase: 'string',
            width: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfoVideoStreamList = BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfoVideoStreamList;
class BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfo extends $dara.Model {
    static names() {
        return {
            audioStreamList: 'AudioStreamList',
            bitrate: 'Bitrate',
            creationTime: 'CreationTime',
            duration: 'Duration',
            fileMD5: 'FileMD5',
            fileName: 'FileName',
            fileURL: 'FileURL',
            fps: 'Fps',
            height: 'Height',
            mediaId: 'MediaId',
            size: 'Size',
            status: 'Status',
            videoStreamList: 'VideoStreamList',
            width: 'Width',
        };
    }
    static types() {
        return {
            audioStreamList: { 'type': 'array', 'itemType': BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfoAudioStreamList },
            bitrate: 'string',
            creationTime: 'string',
            duration: 'string',
            fileMD5: 'string',
            fileName: 'string',
            fileURL: 'string',
            fps: 'string',
            height: 'number',
            mediaId: 'string',
            size: 'number',
            status: 'string',
            videoStreamList: { 'type': 'array', 'itemType': BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfoVideoStreamList },
            width: 'number',
        };
    }
    validate() {
        if (Array.isArray(this.audioStreamList)) {
            $dara.Model.validateArray(this.audioStreamList);
        }
        if (Array.isArray(this.videoStreamList)) {
            $dara.Model.validateArray(this.videoStreamList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfo = BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfo;
class BatchGetMediaInfosResponseBodyMediaInfosPlayInfoList extends $dara.Model {
    static names() {
        return {
            bitDepth: 'BitDepth',
            bitrate: 'Bitrate',
            codecName: 'CodecName',
            creationTime: 'CreationTime',
            definition: 'Definition',
            duration: 'Duration',
            encrypt: 'Encrypt',
            encryptMode: 'EncryptMode',
            encryptType: 'EncryptType',
            format: 'Format',
            fps: 'Fps',
            HDRType: 'HDRType',
            height: 'Height',
            jobExt: 'JobExt',
            jobId: 'JobId',
            jobType: 'JobType',
            modificationTime: 'ModificationTime',
            narrowBandType: 'NarrowBandType',
            playURL: 'PlayURL',
            size: 'Size',
            specification: 'Specification',
            status: 'Status',
            streamType: 'StreamType',
            templateGroupId: 'TemplateGroupId',
            templateId: 'TemplateId',
            watermarkId: 'WatermarkId',
            width: 'Width',
        };
    }
    static types() {
        return {
            bitDepth: 'number',
            bitrate: 'string',
            codecName: 'string',
            creationTime: 'string',
            definition: 'string',
            duration: 'string',
            encrypt: 'number',
            encryptMode: 'string',
            encryptType: 'string',
            format: 'string',
            fps: 'string',
            HDRType: 'string',
            height: 'number',
            jobExt: 'string',
            jobId: 'string',
            jobType: 'number',
            modificationTime: 'string',
            narrowBandType: 'string',
            playURL: 'string',
            size: 'number',
            specification: 'string',
            status: 'string',
            streamType: 'string',
            templateGroupId: 'string',
            templateId: 'string',
            watermarkId: 'string',
            width: 'number',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.BatchGetMediaInfosResponseBodyMediaInfosPlayInfoList = BatchGetMediaInfosResponseBodyMediaInfosPlayInfoList;
class BatchGetMediaInfosResponseBodyMediaInfos extends $dara.Model {
    static names() {
        return {
            mediaId: 'MediaId',
            mediaInfo: 'MediaInfo',
            mezzanineInfo: 'MezzanineInfo',
            playInfoList: 'PlayInfoList',
        };
    }
    static types() {
        return {
            mediaId: 'string',
            mediaInfo: BatchGetMediaInfosResponseBodyMediaInfosMediaInfo,
            mezzanineInfo: BatchGetMediaInfosResponseBodyMediaInfosMezzanineInfo,
            playInfoList: { 'type': 'array', 'itemType': BatchGetMediaInfosResponseBodyMediaInfosPlayInfoList },
        };
    }
    validate() {
        if (this.mediaInfo && typeof this.mediaInfo.validate === 'function') {
            this.mediaInfo.validate();
        }
        if (this.mezzanineInfo && typeof this.mezzanineInfo.validate === 'function') {
            this.mezzanineInfo.validate();
        }
        if (Array.isArray(this.playInfoList)) {
            $dara.Model.validateArray(this.playInfoList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.BatchGetMediaInfosResponseBodyMediaInfos = BatchGetMediaInfosResponseBodyMediaInfos;
class BatchGetMediaInfosResponseBody extends $dara.Model {
    static names() {
        return {
            forbiddenMediaIds: 'ForbiddenMediaIds',
            mediaInfos: 'MediaInfos',
            nonExistMediaIds: 'NonExistMediaIds',
            nonExistReferenceIds: 'NonExistReferenceIds',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            forbiddenMediaIds: { 'type': 'array', 'itemType': 'string' },
            mediaInfos: { 'type': 'array', 'itemType': BatchGetMediaInfosResponseBodyMediaInfos },
            nonExistMediaIds: { 'type': 'array', 'itemType': 'string' },
            nonExistReferenceIds: { 'type': 'array', 'itemType': 'string' },
            requestId: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.forbiddenMediaIds)) {
            $dara.Model.validateArray(this.forbiddenMediaIds);
        }
        if (Array.isArray(this.mediaInfos)) {
            $dara.Model.validateArray(this.mediaInfos);
        }
        if (Array.isArray(this.nonExistMediaIds)) {
            $dara.Model.validateArray(this.nonExistMediaIds);
        }
        if (Array.isArray(this.nonExistReferenceIds)) {
            $dara.Model.validateArray(this.nonExistReferenceIds);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.BatchGetMediaInfosResponseBody = BatchGetMediaInfosResponseBody;
