// @alicloud/vod20170321@3.11.0/dist/models/DescribeVodDomainDetailResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodDomainDetailResponseBody = exports.DescribeVodDomainDetailResponseBodyDomainDetail = exports.DescribeVodDomainDetailResponseBodyDomainDetailSources = exports.DescribeVodDomainDetailResponseBodyDomainDetailSourcesSource = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodDomainDetailResponseBodyDomainDetailSourcesSource extends $dara.Model {
    static names() {
        return {
            content: 'Content',
            enabled: 'Enabled',
            port: 'Port',
            priority: 'Priority',
            type: 'Type',
            weight: 'Weight',
        };
    }
    static types() {
        return {
            content: 'string',
            enabled: 'string',
            port: 'number',
            priority: 'string',
            type: 'string',
            weight: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainDetailResponseBodyDomainDetailSourcesSource = DescribeVodDomainDetailResponseBodyDomainDetailSourcesSource;
class DescribeVodDomainDetailResponseBodyDomainDetailSources extends $dara.Model {
    static names() {
        return {
            source: 'Source',
        };
    }
    static types() {
        return {
            source: { 'type': 'array', 'itemType': DescribeVodDomainDetailResponseBodyDomainDetailSourcesSource },
        };
    }
    validate() {
        if (Array.isArray(this.source)) {
            $dara.Model.validateArray(this.source);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainDetailResponseBodyDomainDetailSources = DescribeVodDomainDetailResponseBodyDomainDetailSources;
class DescribeVodDomainDetailResponseBodyDomainDetail extends $dara.Model {
    static names() {
        return {
            certName: 'CertName',
            cname: 'Cname',
            description: 'Description',
            domainName: 'DomainName',
            domainStatus: 'DomainStatus',
            gmtCreated: 'GmtCreated',
            gmtModified: 'GmtModified',
            SSLProtocol: 'SSLProtocol',
            SSLPub: 'SSLPub',
            scope: 'Scope',
            sources: 'Sources',
            weight: 'Weight',
        };
    }
    static types() {
        return {
            certName: 'string',
            cname: 'string',
            description: 'string',
            domainName: 'string',
            domainStatus: 'string',
            gmtCreated: 'string',
            gmtModified: 'string',
            SSLProtocol: 'string',
            SSLPub: 'string',
            scope: 'string',
            sources: DescribeVodDomainDetailResponseBodyDomainDetailSources,
            weight: 'string',
        };
    }
    validate() {
        if (this.sources && typeof this.sources.validate === 'function') {
            this.sources.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainDetailResponseBodyDomainDetail = DescribeVodDomainDetailResponseBodyDomainDetail;
class DescribeVodDomainDetailResponseBody extends $dara.Model {
    static names() {
        return {
            domainDetail: 'DomainDetail',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            domainDetail: DescribeVodDomainDetailResponseBodyDomainDetail,
            requestId: 'string',
        };
    }
    validate() {
        if (this.domainDetail && typeof this.domainDetail.validate === 'function') {
            this.domainDetail.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainDetailResponseBody = DescribeVodDomainDetailResponseBody;
