// @alicloud/vod20170321@3.11.0/dist/models/DescribeVodDomainUsageDataResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodDomainUsageDataResponseBody = exports.DescribeVodDomainUsageDataResponseBodyUsageDataPerInterval = exports.DescribeVodDomainUsageDataResponseBodyUsageDataPerIntervalDataModule = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodDomainUsageDataResponseBodyUsageDataPerIntervalDataModule extends $dara.Model {
    static names() {
        return {
            timeStamp: 'TimeStamp',
            value: 'Value',
        };
    }
    static types() {
        return {
            timeStamp: 'string',
            value: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainUsageDataResponseBodyUsageDataPerIntervalDataModule = DescribeVodDomainUsageDataResponseBodyUsageDataPerIntervalDataModule;
class DescribeVodDomainUsageDataResponseBodyUsageDataPerInterval extends $dara.Model {
    static names() {
        return {
            dataModule: 'DataModule',
        };
    }
    static types() {
        return {
            dataModule: { 'type': 'array', 'itemType': DescribeVodDomainUsageDataResponseBodyUsageDataPerIntervalDataModule },
        };
    }
    validate() {
        if (Array.isArray(this.dataModule)) {
            $dara.Model.validateArray(this.dataModule);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainUsageDataResponseBodyUsageDataPerInterval = DescribeVodDomainUsageDataResponseBodyUsageDataPerInterval;
class DescribeVodDomainUsageDataResponseBody extends $dara.Model {
    static names() {
        return {
            area: 'Area',
            dataInterval: 'DataInterval',
            domainName: 'DomainName',
            endTime: 'EndTime',
            requestId: 'RequestId',
            startTime: 'StartTime',
            type: 'Type',
            usageDataPerInterval: 'UsageDataPerInterval',
        };
    }
    static types() {
        return {
            area: 'string',
            dataInterval: 'string',
            domainName: 'string',
            endTime: 'string',
            requestId: 'string',
            startTime: 'string',
            type: 'string',
            usageDataPerInterval: DescribeVodDomainUsageDataResponseBodyUsageDataPerInterval,
        };
    }
    validate() {
        if (this.usageDataPerInterval && typeof this.usageDataPerInterval.validate === 'function') {
            this.usageDataPerInterval.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainUsageDataResponseBody = DescribeVodDomainUsageDataResponseBody;
