// @alicloud/vod20170321@3.11.0/dist/models/DescribeVodPlayerMetricDataResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodPlayerMetricDataResponseBody = exports.DescribeVodPlayerMetricDataResponseBodyExtend = exports.DescribeVodPlayerMetricDataResponseBodyDataList = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodPlayerMetricDataResponseBodyDataList extends $dara.Model {
    static names() {
        return {
            avgPerCompletionVv: 'AvgPerCompletionVv',
            avgPerPlayDuration: 'AvgPerPlayDuration',
            avgPerVv: 'AvgPerVv',
            avgPlayBitrate: 'AvgPlayBitrate',
            avgPlayDuration: 'AvgPlayDuration',
            avgStartBitrate: 'AvgStartBitrate',
            avgVideoDuration: 'AvgVideoDuration',
            completionRate: 'CompletionRate',
            completionVv: 'CompletionVv',
            dimension: 'Dimension',
            errorCount100s: 'ErrorCount100s',
            firstFrame: 'FirstFrame',
            jumpRate5s: 'JumpRate5s',
            playFailRate: 'PlayFailRate',
            realVv: 'RealVv',
            secondPlayRate: 'SecondPlayRate',
            seedFailRate: 'SeedFailRate',
            seekDuration: 'SeekDuration',
            slowPlayRate: 'SlowPlayRate',
            stuckCount100s: 'StuckCount100s',
            stuckCountRate: 'StuckCountRate',
            stuckDuration100s: 'StuckDuration100s',
            timeStamp: 'TimeStamp',
            totalPlayDuration: 'TotalPlayDuration',
            uv: 'Uv',
            vv: 'Vv',
        };
    }
    static types() {
        return {
            avgPerCompletionVv: 'number',
            avgPerPlayDuration: 'number',
            avgPerVv: 'number',
            avgPlayBitrate: 'number',
            avgPlayDuration: 'number',
            avgStartBitrate: 'number',
            avgVideoDuration: 'number',
            completionRate: 'number',
            completionVv: 'number',
            dimension: 'string',
            errorCount100s: 'number',
            firstFrame: 'number',
            jumpRate5s: 'number',
            playFailRate: 'number',
            realVv: 'number',
            secondPlayRate: 'number',
            seedFailRate: 'number',
            seekDuration: 'number',
            slowPlayRate: 'number',
            stuckCount100s: 'string',
            stuckCountRate: 'number',
            stuckDuration100s: 'number',
            timeStamp: 'string',
            totalPlayDuration: 'number',
            uv: 'number',
            vv: 'number',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodPlayerMetricDataResponseBodyDataList = DescribeVodPlayerMetricDataResponseBodyDataList;
class DescribeVodPlayerMetricDataResponseBodyExtend extends $dara.Model {
    static names() {
        return {
            actualEndTime: 'ActualEndTime',
            actualStartTime: 'ActualStartTime',
            intervalSeconds: 'IntervalSeconds',
        };
    }
    static types() {
        return {
            actualEndTime: 'string',
            actualStartTime: 'string',
            intervalSeconds: 'number',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodPlayerMetricDataResponseBodyExtend = DescribeVodPlayerMetricDataResponseBodyExtend;
class DescribeVodPlayerMetricDataResponseBody extends $dara.Model {
    static names() {
        return {
            dataList: 'DataList',
            extend: 'Extend',
            pageNumber: 'PageNumber',
            pageSize: 'PageSize',
            requestId: 'RequestId',
            totalCnt: 'TotalCnt',
        };
    }
    static types() {
        return {
            dataList: { 'type': 'array', 'itemType': DescribeVodPlayerMetricDataResponseBodyDataList },
            extend: DescribeVodPlayerMetricDataResponseBodyExtend,
            pageNumber: 'number',
            pageSize: 'number',
            requestId: 'string',
            totalCnt: 'number',
        };
    }
    validate() {
        if (Array.isArray(this.dataList)) {
            $dara.Model.validateArray(this.dataList);
        }
        if (this.extend && typeof this.extend.validate === 'function') {
            this.extend.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodPlayerMetricDataResponseBody = DescribeVodPlayerMetricDataResponseBody;
