// @alicloud/vod20170321@3.11.0/dist/models/DescribeVodStorageDataResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodStorageDataResponseBody = exports.DescribeVodStorageDataResponseBodyStorageData = exports.DescribeVodStorageDataResponseBodyStorageDataStorageDataItem = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodStorageDataResponseBodyStorageDataStorageDataItem extends $dara.Model {
    static names() {
        return {
            networkOut: 'NetworkOut',
            storageUtilization: 'StorageUtilization',
            timeStamp: 'TimeStamp',
        };
    }
    static types() {
        return {
            networkOut: 'string',
            storageUtilization: 'string',
            timeStamp: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodStorageDataResponseBodyStorageDataStorageDataItem = DescribeVodStorageDataResponseBodyStorageDataStorageDataItem;
class DescribeVodStorageDataResponseBodyStorageData extends $dara.Model {
    static names() {
        return {
            storageDataItem: 'StorageDataItem',
        };
    }
    static types() {
        return {
            storageDataItem: { 'type': 'array', 'itemType': DescribeVodStorageDataResponseBodyStorageDataStorageDataItem },
        };
    }
    validate() {
        if (Array.isArray(this.storageDataItem)) {
            $dara.Model.validateArray(this.storageDataItem);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodStorageDataResponseBodyStorageData = DescribeVodStorageDataResponseBodyStorageData;
class DescribeVodStorageDataResponseBody extends $dara.Model {
    static names() {
        return {
            dataInterval: 'DataInterval',
            requestId: 'RequestId',
            storageData: 'StorageData',
        };
    }
    static types() {
        return {
            dataInterval: 'string',
            requestId: 'string',
            storageData: DescribeVodStorageDataResponseBodyStorageData,
        };
    }
    validate() {
        if (this.storageData && typeof this.storageData.validate === 'function') {
            this.storageData.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodStorageDataResponseBody = DescribeVodStorageDataResponseBody;
