// @alicloud/vod20170321@3.11.0/dist/models/DescribeVodDomainConfigsResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodDomainConfigsResponseBody = exports.DescribeVodDomainConfigsResponseBodyDomainConfigs = exports.DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfig = exports.DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfigFunctionArgs = exports.DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfigFunctionArgsFunctionArg = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfigFunctionArgsFunctionArg extends $dara.Model {
    static names() {
        return {
            argName: 'ArgName',
            argValue: 'ArgValue',
        };
    }
    static types() {
        return {
            argName: 'string',
            argValue: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfigFunctionArgsFunctionArg = DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfigFunctionArgsFunctionArg;
class DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfigFunctionArgs extends $dara.Model {
    static names() {
        return {
            functionArg: 'FunctionArg',
        };
    }
    static types() {
        return {
            functionArg: { 'type': 'array', 'itemType': DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfigFunctionArgsFunctionArg },
        };
    }
    validate() {
        if (Array.isArray(this.functionArg)) {
            $dara.Model.validateArray(this.functionArg);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfigFunctionArgs = DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfigFunctionArgs;
class DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfig extends $dara.Model {
    static names() {
        return {
            configId: 'ConfigId',
            functionArgs: 'FunctionArgs',
            functionName: 'FunctionName',
            status: 'Status',
        };
    }
    static types() {
        return {
            configId: 'string',
            functionArgs: DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfigFunctionArgs,
            functionName: 'string',
            status: 'string',
        };
    }
    validate() {
        if (this.functionArgs && typeof this.functionArgs.validate === 'function') {
            this.functionArgs.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfig = DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfig;
class DescribeVodDomainConfigsResponseBodyDomainConfigs extends $dara.Model {
    static names() {
        return {
            domainConfig: 'DomainConfig',
        };
    }
    static types() {
        return {
            domainConfig: { 'type': 'array', 'itemType': DescribeVodDomainConfigsResponseBodyDomainConfigsDomainConfig },
        };
    }
    validate() {
        if (Array.isArray(this.domainConfig)) {
            $dara.Model.validateArray(this.domainConfig);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainConfigsResponseBodyDomainConfigs = DescribeVodDomainConfigsResponseBodyDomainConfigs;
class DescribeVodDomainConfigsResponseBody extends $dara.Model {
    static names() {
        return {
            domainConfigs: 'DomainConfigs',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            domainConfigs: DescribeVodDomainConfigsResponseBodyDomainConfigs,
            requestId: 'string',
        };
    }
    validate() {
        if (this.domainConfigs && typeof this.domainConfigs.validate === 'function') {
            this.domainConfigs.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainConfigsResponseBody = DescribeVodDomainConfigsResponseBody;
