// @alicloud/vod20170321@3.11.0/dist/models/DescribeVodDomainRealTimeHttpCodeDataResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodDomainRealTimeHttpCodeDataResponseBody = exports.DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeData = exports.DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageData = exports.DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageDataValue = exports.DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageDataValueRealTimeCodeProportionData = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageDataValueRealTimeCodeProportionData extends $dara.Model {
    static names() {
        return {
            code: 'Code',
            count: 'Count',
            proportion: 'Proportion',
        };
    }
    static types() {
        return {
            code: 'string',
            count: 'string',
            proportion: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageDataValueRealTimeCodeProportionData = DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageDataValueRealTimeCodeProportionData;
class DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageDataValue extends $dara.Model {
    static names() {
        return {
            realTimeCodeProportionData: 'RealTimeCodeProportionData',
        };
    }
    static types() {
        return {
            realTimeCodeProportionData: { 'type': 'array', 'itemType': DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageDataValueRealTimeCodeProportionData },
        };
    }
    validate() {
        if (Array.isArray(this.realTimeCodeProportionData)) {
            $dara.Model.validateArray(this.realTimeCodeProportionData);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageDataValue = DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageDataValue;
class DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageData extends $dara.Model {
    static names() {
        return {
            timeStamp: 'TimeStamp',
            value: 'Value',
        };
    }
    static types() {
        return {
            timeStamp: 'string',
            value: DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageDataValue,
        };
    }
    validate() {
        if (this.value && typeof this.value.validate === 'function') {
            this.value.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageData = DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageData;
class DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeData extends $dara.Model {
    static names() {
        return {
            usageData: 'UsageData',
        };
    }
    static types() {
        return {
            usageData: { 'type': 'array', 'itemType': DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeDataUsageData },
        };
    }
    validate() {
        if (Array.isArray(this.usageData)) {
            $dara.Model.validateArray(this.usageData);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeData = DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeData;
class DescribeVodDomainRealTimeHttpCodeDataResponseBody extends $dara.Model {
    static names() {
        return {
            dataInterval: 'DataInterval',
            domainName: 'DomainName',
            endTime: 'EndTime',
            realTimeHttpCodeData: 'RealTimeHttpCodeData',
            requestId: 'RequestId',
            startTime: 'StartTime',
        };
    }
    static types() {
        return {
            dataInterval: 'string',
            domainName: 'string',
            endTime: 'string',
            realTimeHttpCodeData: DescribeVodDomainRealTimeHttpCodeDataResponseBodyRealTimeHttpCodeData,
            requestId: 'string',
            startTime: 'string',
        };
    }
    validate() {
        if (this.realTimeHttpCodeData && typeof this.realTimeHttpCodeData.validate === 'function') {
            this.realTimeHttpCodeData.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainRealTimeHttpCodeDataResponseBody = DescribeVodDomainRealTimeHttpCodeDataResponseBody;
