// @alicloud/vod20170321@3.11.0/dist/models/DescribeVodAidataResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodAIDataResponseBody = exports.DescribeVodAIDataResponseBodyAIData = exports.DescribeVodAIDataResponseBodyAIDataAIDataItem = exports.DescribeVodAIDataResponseBodyAIDataAIDataItemData = exports.DescribeVodAIDataResponseBodyAIDataAIDataItemDataDataItem = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodAIDataResponseBodyAIDataAIDataItemDataDataItem extends $dara.Model {
    static names() {
        return {
            name: 'Name',
            value: 'Value',
        };
    }
    static types() {
        return {
            name: 'string',
            value: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodAIDataResponseBodyAIDataAIDataItemDataDataItem = DescribeVodAIDataResponseBodyAIDataAIDataItemDataDataItem;
class DescribeVodAIDataResponseBodyAIDataAIDataItemData extends $dara.Model {
    static names() {
        return {
            dataItem: 'DataItem',
        };
    }
    static types() {
        return {
            dataItem: { 'type': 'array', 'itemType': DescribeVodAIDataResponseBodyAIDataAIDataItemDataDataItem },
        };
    }
    validate() {
        if (Array.isArray(this.dataItem)) {
            $dara.Model.validateArray(this.dataItem);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodAIDataResponseBodyAIDataAIDataItemData = DescribeVodAIDataResponseBodyAIDataAIDataItemData;
class DescribeVodAIDataResponseBodyAIDataAIDataItem extends $dara.Model {
    static names() {
        return {
            data: 'Data',
            timeStamp: 'TimeStamp',
        };
    }
    static types() {
        return {
            data: DescribeVodAIDataResponseBodyAIDataAIDataItemData,
            timeStamp: 'string',
        };
    }
    validate() {
        if (this.data && typeof this.data.validate === 'function') {
            this.data.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodAIDataResponseBodyAIDataAIDataItem = DescribeVodAIDataResponseBodyAIDataAIDataItem;
class DescribeVodAIDataResponseBodyAIData extends $dara.Model {
    static names() {
        return {
            AIDataItem: 'AIDataItem',
        };
    }
    static types() {
        return {
            AIDataItem: { 'type': 'array', 'itemType': DescribeVodAIDataResponseBodyAIDataAIDataItem },
        };
    }
    validate() {
        if (Array.isArray(this.AIDataItem)) {
            $dara.Model.validateArray(this.AIDataItem);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodAIDataResponseBodyAIData = DescribeVodAIDataResponseBodyAIData;
class DescribeVodAIDataResponseBody extends $dara.Model {
    static names() {
        return {
            AIData: 'AIData',
            dataInterval: 'DataInterval',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            AIData: DescribeVodAIDataResponseBodyAIData,
            dataInterval: 'string',
            requestId: 'string',
        };
    }
    validate() {
        if (this.AIData && typeof this.AIData.validate === 'function') {
            this.AIData.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodAIDataResponseBody = DescribeVodAIDataResponseBody;
