// @alicloud/vod20170321@3.11.0/dist/models/DescribePlayUserTotalResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribePlayUserTotalResponseBody = exports.DescribePlayUserTotalResponseBodyUserPlayStatisTotals = exports.DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotal = exports.DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotalVV = exports.DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotalUV = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotalUV extends $dara.Model {
    static names() {
        return {
            android: 'Android',
            flash: 'Flash',
            HTML5: 'HTML5',
            iOS: 'iOS',
        };
    }
    static types() {
        return {
            android: 'string',
            flash: 'string',
            HTML5: 'string',
            iOS: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotalUV = DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotalUV;
class DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotalVV extends $dara.Model {
    static names() {
        return {
            android: 'Android',
            flash: 'Flash',
            HTML5: 'HTML5',
            iOS: 'iOS',
        };
    }
    static types() {
        return {
            android: 'string',
            flash: 'string',
            HTML5: 'string',
            iOS: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotalVV = DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotalVV;
class DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotal extends $dara.Model {
    static names() {
        return {
            date: 'Date',
            playDuration: 'PlayDuration',
            playRange: 'PlayRange',
            UV: 'UV',
            VV: 'VV',
        };
    }
    static types() {
        return {
            date: 'string',
            playDuration: 'string',
            playRange: 'string',
            UV: DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotalUV,
            VV: DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotalVV,
        };
    }
    validate() {
        if (this.UV && typeof this.UV.validate === 'function') {
            this.UV.validate();
        }
        if (this.VV && typeof this.VV.validate === 'function') {
            this.VV.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotal = DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotal;
class DescribePlayUserTotalResponseBodyUserPlayStatisTotals extends $dara.Model {
    static names() {
        return {
            userPlayStatisTotal: 'UserPlayStatisTotal',
        };
    }
    static types() {
        return {
            userPlayStatisTotal: { 'type': 'array', 'itemType': DescribePlayUserTotalResponseBodyUserPlayStatisTotalsUserPlayStatisTotal },
        };
    }
    validate() {
        if (Array.isArray(this.userPlayStatisTotal)) {
            $dara.Model.validateArray(this.userPlayStatisTotal);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribePlayUserTotalResponseBodyUserPlayStatisTotals = DescribePlayUserTotalResponseBodyUserPlayStatisTotals;
class DescribePlayUserTotalResponseBody extends $dara.Model {
    static names() {
        return {
            requestId: 'RequestId',
            userPlayStatisTotals: 'UserPlayStatisTotals',
        };
    }
    static types() {
        return {
            requestId: 'string',
            userPlayStatisTotals: DescribePlayUserTotalResponseBodyUserPlayStatisTotals,
        };
    }
    validate() {
        if (this.userPlayStatisTotals && typeof this.userPlayStatisTotals.validate === 'function') {
            this.userPlayStatisTotals.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribePlayUserTotalResponseBody = DescribePlayUserTotalResponseBody;
