// @alicloud/vod20170321@3.11.0/dist/models/DescribeVodDomainMax95bpsDataResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodDomainMax95BpsDataResponseBody = exports.DescribeVodDomainMax95BpsDataResponseBodyDetailData = exports.DescribeVodDomainMax95BpsDataResponseBodyDetailDataMax95Detail = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodDomainMax95BpsDataResponseBodyDetailDataMax95Detail extends $dara.Model {
    static names() {
        return {
            area: 'Area',
            max95Bps: 'Max95Bps',
            max95BpsPeakTime: 'Max95BpsPeakTime',
            timeStamp: 'TimeStamp',
        };
    }
    static types() {
        return {
            area: 'string',
            max95Bps: 'number',
            max95BpsPeakTime: 'string',
            timeStamp: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainMax95BpsDataResponseBodyDetailDataMax95Detail = DescribeVodDomainMax95BpsDataResponseBodyDetailDataMax95Detail;
class DescribeVodDomainMax95BpsDataResponseBodyDetailData extends $dara.Model {
    static names() {
        return {
            max95Detail: 'Max95Detail',
        };
    }
    static types() {
        return {
            max95Detail: { 'type': 'array', 'itemType': DescribeVodDomainMax95BpsDataResponseBodyDetailDataMax95Detail },
        };
    }
    validate() {
        if (Array.isArray(this.max95Detail)) {
            $dara.Model.validateArray(this.max95Detail);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainMax95BpsDataResponseBodyDetailData = DescribeVodDomainMax95BpsDataResponseBodyDetailData;
class DescribeVodDomainMax95BpsDataResponseBody extends $dara.Model {
    static names() {
        return {
            detailData: 'DetailData',
            domainName: 'DomainName',
            domesticMax95Bps: 'DomesticMax95Bps',
            endTime: 'EndTime',
            max95Bps: 'Max95Bps',
            overseasMax95Bps: 'OverseasMax95Bps',
            requestId: 'RequestId',
            startTime: 'StartTime',
        };
    }
    static types() {
        return {
            detailData: DescribeVodDomainMax95BpsDataResponseBodyDetailData,
            domainName: 'string',
            domesticMax95Bps: 'string',
            endTime: 'string',
            max95Bps: 'string',
            overseasMax95Bps: 'string',
            requestId: 'string',
            startTime: 'string',
        };
    }
    validate() {
        if (this.detailData && typeof this.detailData.validate === 'function') {
            this.detailData.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainMax95BpsDataResponseBody = DescribeVodDomainMax95BpsDataResponseBody;
