// @alicloud/vod20170321@3.11.0/dist/models/DescribeVodSslcertificateListResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodSSLCertificateListResponseBody = exports.DescribeVodSSLCertificateListResponseBodyCertificateListModel = exports.DescribeVodSSLCertificateListResponseBodyCertificateListModelCertList = exports.DescribeVodSSLCertificateListResponseBodyCertificateListModelCertListCert = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodSSLCertificateListResponseBodyCertificateListModelCertListCert extends $dara.Model {
    static names() {
        return {
            certId: 'CertId',
            certName: 'CertName',
            certRegion: 'CertRegion',
            common: 'Common',
            fingerprint: 'Fingerprint',
            issuer: 'Issuer',
            lastTime: 'LastTime',
        };
    }
    static types() {
        return {
            certId: 'number',
            certName: 'string',
            certRegion: 'string',
            common: 'string',
            fingerprint: 'string',
            issuer: 'string',
            lastTime: 'number',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodSSLCertificateListResponseBodyCertificateListModelCertListCert = DescribeVodSSLCertificateListResponseBodyCertificateListModelCertListCert;
class DescribeVodSSLCertificateListResponseBodyCertificateListModelCertList extends $dara.Model {
    static names() {
        return {
            cert: 'Cert',
        };
    }
    static types() {
        return {
            cert: { 'type': 'array', 'itemType': DescribeVodSSLCertificateListResponseBodyCertificateListModelCertListCert },
        };
    }
    validate() {
        if (Array.isArray(this.cert)) {
            $dara.Model.validateArray(this.cert);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodSSLCertificateListResponseBodyCertificateListModelCertList = DescribeVodSSLCertificateListResponseBodyCertificateListModelCertList;
class DescribeVodSSLCertificateListResponseBodyCertificateListModel extends $dara.Model {
    static names() {
        return {
            certList: 'CertList',
            count: 'Count',
            pageNumber: 'PageNumber',
            pageSize: 'PageSize',
        };
    }
    static types() {
        return {
            certList: DescribeVodSSLCertificateListResponseBodyCertificateListModelCertList,
            count: 'number',
            pageNumber: 'number',
            pageSize: 'number',
        };
    }
    validate() {
        if (this.certList && typeof this.certList.validate === 'function') {
            this.certList.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodSSLCertificateListResponseBodyCertificateListModel = DescribeVodSSLCertificateListResponseBodyCertificateListModel;
class DescribeVodSSLCertificateListResponseBody extends $dara.Model {
    static names() {
        return {
            certificateListModel: 'CertificateListModel',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            certificateListModel: DescribeVodSSLCertificateListResponseBodyCertificateListModel,
            requestId: 'string',
        };
    }
    validate() {
        if (this.certificateListModel && typeof this.certificateListModel.validate === 'function') {
            this.certificateListModel.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodSSLCertificateListResponseBody = DescribeVodSSLCertificateListResponseBody;
