// @alicloud/vod20170321@3.11.0/dist/models/DescribeVodDomainLogResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodDomainLogResponseBody = exports.DescribeVodDomainLogResponseBodyDomainLogDetails = exports.DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetail = exports.DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailPageInfos = exports.DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailLogInfos = exports.DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailLogInfosLogInfoDetail = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailLogInfosLogInfoDetail extends $dara.Model {
    static names() {
        return {
            endTime: 'EndTime',
            logName: 'LogName',
            logPath: 'LogPath',
            logSize: 'LogSize',
            startTime: 'StartTime',
        };
    }
    static types() {
        return {
            endTime: 'string',
            logName: 'string',
            logPath: 'string',
            logSize: 'number',
            startTime: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailLogInfosLogInfoDetail = DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailLogInfosLogInfoDetail;
class DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailLogInfos extends $dara.Model {
    static names() {
        return {
            logInfoDetail: 'LogInfoDetail',
        };
    }
    static types() {
        return {
            logInfoDetail: { 'type': 'array', 'itemType': DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailLogInfosLogInfoDetail },
        };
    }
    validate() {
        if (Array.isArray(this.logInfoDetail)) {
            $dara.Model.validateArray(this.logInfoDetail);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailLogInfos = DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailLogInfos;
class DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailPageInfos extends $dara.Model {
    static names() {
        return {
            pageNumber: 'PageNumber',
            pageSize: 'PageSize',
            total: 'Total',
        };
    }
    static types() {
        return {
            pageNumber: 'number',
            pageSize: 'number',
            total: 'number',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailPageInfos = DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailPageInfos;
class DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetail extends $dara.Model {
    static names() {
        return {
            domainName: 'DomainName',
            logCount: 'LogCount',
            logInfos: 'LogInfos',
            pageInfos: 'PageInfos',
        };
    }
    static types() {
        return {
            domainName: 'string',
            logCount: 'number',
            logInfos: DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailLogInfos,
            pageInfos: DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetailPageInfos,
        };
    }
    validate() {
        if (this.logInfos && typeof this.logInfos.validate === 'function') {
            this.logInfos.validate();
        }
        if (this.pageInfos && typeof this.pageInfos.validate === 'function') {
            this.pageInfos.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetail = DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetail;
class DescribeVodDomainLogResponseBodyDomainLogDetails extends $dara.Model {
    static names() {
        return {
            domainLogDetail: 'DomainLogDetail',
        };
    }
    static types() {
        return {
            domainLogDetail: { 'type': 'array', 'itemType': DescribeVodDomainLogResponseBodyDomainLogDetailsDomainLogDetail },
        };
    }
    validate() {
        if (Array.isArray(this.domainLogDetail)) {
            $dara.Model.validateArray(this.domainLogDetail);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainLogResponseBodyDomainLogDetails = DescribeVodDomainLogResponseBodyDomainLogDetails;
class DescribeVodDomainLogResponseBody extends $dara.Model {
    static names() {
        return {
            domainLogDetails: 'DomainLogDetails',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            domainLogDetails: DescribeVodDomainLogResponseBodyDomainLogDetails,
            requestId: 'string',
        };
    }
    validate() {
        if (this.domainLogDetails && typeof this.domainLogDetails.validate === 'function') {
            this.domainLogDetails.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainLogResponseBody = DescribeVodDomainLogResponseBody;
