// @alicloud/vod20170321@3.11.0/dist/models/DescribeVodDomainRealTimeByteHitRateDataResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeVodDomainRealTimeByteHitRateDataResponseBody = exports.DescribeVodDomainRealTimeByteHitRateDataResponseBodyData = exports.DescribeVodDomainRealTimeByteHitRateDataResponseBodyDataByteHitRateDataModel = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class DescribeVodDomainRealTimeByteHitRateDataResponseBodyDataByteHitRateDataModel extends $dara.Model {
    static names() {
        return {
            byteHitRate: 'ByteHitRate',
            timeStamp: 'TimeStamp',
        };
    }
    static types() {
        return {
            byteHitRate: 'number',
            timeStamp: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainRealTimeByteHitRateDataResponseBodyDataByteHitRateDataModel = DescribeVodDomainRealTimeByteHitRateDataResponseBodyDataByteHitRateDataModel;
class DescribeVodDomainRealTimeByteHitRateDataResponseBodyData extends $dara.Model {
    static names() {
        return {
            byteHitRateDataModel: 'ByteHitRateDataModel',
        };
    }
    static types() {
        return {
            byteHitRateDataModel: { 'type': 'array', 'itemType': DescribeVodDomainRealTimeByteHitRateDataResponseBodyDataByteHitRateDataModel },
        };
    }
    validate() {
        if (Array.isArray(this.byteHitRateDataModel)) {
            $dara.Model.validateArray(this.byteHitRateDataModel);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainRealTimeByteHitRateDataResponseBodyData = DescribeVodDomainRealTimeByteHitRateDataResponseBodyData;
class DescribeVodDomainRealTimeByteHitRateDataResponseBody extends $dara.Model {
    static names() {
        return {
            data: 'Data',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            data: DescribeVodDomainRealTimeByteHitRateDataResponseBodyData,
            requestId: 'string',
        };
    }
    validate() {
        if (this.data && typeof this.data.validate === 'function') {
            this.data.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.DescribeVodDomainRealTimeByteHitRateDataResponseBody = DescribeVodDomainRealTimeByteHitRateDataResponseBody;
