// @alicloud/vod20170321@3.11.0/dist/models/GetEditingProjectMaterialsResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetEditingProjectMaterialsResponseBody = exports.GetEditingProjectMaterialsResponseBodyMaterialList = exports.GetEditingProjectMaterialsResponseBodyMaterialListMaterial = exports.GetEditingProjectMaterialsResponseBodyMaterialListMaterialSprites = exports.GetEditingProjectMaterialsResponseBodyMaterialListMaterialSnapshots = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetEditingProjectMaterialsResponseBodyMaterialListMaterialSnapshots extends $dara.Model {
    static names() {
        return {
            snapshot: 'Snapshot',
        };
    }
    static types() {
        return {
            snapshot: { 'type': 'array', 'itemType': 'string' },
        };
    }
    validate() {
        if (Array.isArray(this.snapshot)) {
            $dara.Model.validateArray(this.snapshot);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetEditingProjectMaterialsResponseBodyMaterialListMaterialSnapshots = GetEditingProjectMaterialsResponseBodyMaterialListMaterialSnapshots;
class GetEditingProjectMaterialsResponseBodyMaterialListMaterialSprites extends $dara.Model {
    static names() {
        return {
            sprite: 'Sprite',
        };
    }
    static types() {
        return {
            sprite: { 'type': 'array', 'itemType': 'string' },
        };
    }
    validate() {
        if (Array.isArray(this.sprite)) {
            $dara.Model.validateArray(this.sprite);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetEditingProjectMaterialsResponseBodyMaterialListMaterialSprites = GetEditingProjectMaterialsResponseBodyMaterialListMaterialSprites;
class GetEditingProjectMaterialsResponseBodyMaterialListMaterial extends $dara.Model {
    static names() {
        return {
            cateId: 'CateId',
            cateName: 'CateName',
            coverURL: 'CoverURL',
            creationTime: 'CreationTime',
            description: 'Description',
            duration: 'Duration',
            materialId: 'MaterialId',
            materialType: 'MaterialType',
            modifiedTime: 'ModifiedTime',
            size: 'Size',
            snapshots: 'Snapshots',
            source: 'Source',
            spriteConfig: 'SpriteConfig',
            sprites: 'Sprites',
            status: 'Status',
            tags: 'Tags',
            title: 'Title',
        };
    }
    static types() {
        return {
            cateId: 'number',
            cateName: 'string',
            coverURL: 'string',
            creationTime: 'string',
            description: 'string',
            duration: 'number',
            materialId: 'string',
            materialType: 'string',
            modifiedTime: 'string',
            size: 'number',
            snapshots: GetEditingProjectMaterialsResponseBodyMaterialListMaterialSnapshots,
            source: 'string',
            spriteConfig: 'string',
            sprites: GetEditingProjectMaterialsResponseBodyMaterialListMaterialSprites,
            status: 'string',
            tags: 'string',
            title: 'string',
        };
    }
    validate() {
        if (this.snapshots && typeof this.snapshots.validate === 'function') {
            this.snapshots.validate();
        }
        if (this.sprites && typeof this.sprites.validate === 'function') {
            this.sprites.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetEditingProjectMaterialsResponseBodyMaterialListMaterial = GetEditingProjectMaterialsResponseBodyMaterialListMaterial;
class GetEditingProjectMaterialsResponseBodyMaterialList extends $dara.Model {
    static names() {
        return {
            material: 'Material',
        };
    }
    static types() {
        return {
            material: { 'type': 'array', 'itemType': GetEditingProjectMaterialsResponseBodyMaterialListMaterial },
        };
    }
    validate() {
        if (Array.isArray(this.material)) {
            $dara.Model.validateArray(this.material);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetEditingProjectMaterialsResponseBodyMaterialList = GetEditingProjectMaterialsResponseBodyMaterialList;
class GetEditingProjectMaterialsResponseBody extends $dara.Model {
    static names() {
        return {
            materialList: 'MaterialList',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            materialList: GetEditingProjectMaterialsResponseBodyMaterialList,
            requestId: 'string',
        };
    }
    validate() {
        if (this.materialList && typeof this.materialList.validate === 'function') {
            this.materialList.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetEditingProjectMaterialsResponseBody = GetEditingProjectMaterialsResponseBody;
