// @alicloud/vod20170321@3.11.0/dist/models/GetImageInfoResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetImageInfoResponseBody = exports.GetImageInfoResponseBodyImageInfo = exports.GetImageInfoResponseBodyImageInfoMezzanine = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetImageInfoResponseBodyImageInfoMezzanine extends $dara.Model {
    static names() {
        return {
            fileSize: 'FileSize',
            fileURL: 'FileURL',
            height: 'Height',
            originalFileName: 'OriginalFileName',
            width: 'Width',
        };
    }
    static types() {
        return {
            fileSize: 'string',
            fileURL: 'string',
            height: 'number',
            originalFileName: 'string',
            width: 'number',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetImageInfoResponseBodyImageInfoMezzanine = GetImageInfoResponseBodyImageInfoMezzanine;
class GetImageInfoResponseBodyImageInfo extends $dara.Model {
    static names() {
        return {
            appId: 'AppId',
            cateId: 'CateId',
            cateName: 'CateName',
            creationTime: 'CreationTime',
            description: 'Description',
            imageId: 'ImageId',
            imageType: 'ImageType',
            mezzanine: 'Mezzanine',
            status: 'Status',
            storageLocation: 'StorageLocation',
            tags: 'Tags',
            title: 'Title',
            URL: 'URL',
        };
    }
    static types() {
        return {
            appId: 'string',
            cateId: 'number',
            cateName: 'string',
            creationTime: 'string',
            description: 'string',
            imageId: 'string',
            imageType: 'string',
            mezzanine: GetImageInfoResponseBodyImageInfoMezzanine,
            status: 'string',
            storageLocation: 'string',
            tags: 'string',
            title: 'string',
            URL: 'string',
        };
    }
    validate() {
        if (this.mezzanine && typeof this.mezzanine.validate === 'function') {
            this.mezzanine.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetImageInfoResponseBodyImageInfo = GetImageInfoResponseBodyImageInfo;
class GetImageInfoResponseBody extends $dara.Model {
    static names() {
        return {
            imageInfo: 'ImageInfo',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            imageInfo: GetImageInfoResponseBodyImageInfo,
            requestId: 'string',
        };
    }
    validate() {
        if (this.imageInfo && typeof this.imageInfo.validate === 'function') {
            this.imageInfo.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetImageInfoResponseBody = GetImageInfoResponseBody;
