// @alicloud/vod20170321@3.11.0/dist/models/GetJobDetailResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetJobDetailResponseBody = exports.GetJobDetailResponseBodyWorkflowTaskDetail = exports.GetJobDetailResponseBodyWorkflowTaskDetailWorkflow = exports.GetJobDetailResponseBodyTranscodeJobDetail = exports.GetJobDetailResponseBodySnapshotJobDetail = exports.GetJobDetailResponseBodyAIJobDetail = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetJobDetailResponseBodyAIJobDetail extends $dara.Model {
    static names() {
        return {
            completeTime: 'CompleteTime',
            createTime: 'CreateTime',
            jobId: 'JobId',
            jobType: 'JobType',
            mediaId: 'MediaId',
            status: 'Status',
            templateConfig: 'TemplateConfig',
            trigger: 'Trigger',
            userId: 'UserId',
        };
    }
    static types() {
        return {
            completeTime: 'string',
            createTime: 'string',
            jobId: 'string',
            jobType: 'string',
            mediaId: 'string',
            status: 'string',
            templateConfig: 'string',
            trigger: 'string',
            userId: 'number',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetJobDetailResponseBodyAIJobDetail = GetJobDetailResponseBodyAIJobDetail;
class GetJobDetailResponseBodySnapshotJobDetail extends $dara.Model {
    static names() {
        return {
            completeTime: 'CompleteTime',
            createTime: 'CreateTime',
            jobId: 'JobId',
            normalConfig: 'NormalConfig',
            spriteConfig: 'SpriteConfig',
            status: 'Status',
            trigger: 'Trigger',
            userId: 'UserId',
            videoId: 'VideoId',
        };
    }
    static types() {
        return {
            completeTime: 'string',
            createTime: 'string',
            jobId: 'string',
            normalConfig: 'string',
            spriteConfig: 'string',
            status: 'string',
            trigger: 'string',
            userId: 'number',
            videoId: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetJobDetailResponseBodySnapshotJobDetail = GetJobDetailResponseBodySnapshotJobDetail;
class GetJobDetailResponseBodyTranscodeJobDetail extends $dara.Model {
    static names() {
        return {
            completeTime: 'CompleteTime',
            createTime: 'CreateTime',
            definition: 'Definition',
            jobId: 'JobId',
            status: 'Status',
            templateId: 'TemplateId',
            userId: 'UserId',
            videoId: 'VideoId',
        };
    }
    static types() {
        return {
            completeTime: 'string',
            createTime: 'string',
            definition: 'string',
            jobId: 'string',
            status: 'string',
            templateId: 'string',
            userId: 'number',
            videoId: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetJobDetailResponseBodyTranscodeJobDetail = GetJobDetailResponseBodyTranscodeJobDetail;
class GetJobDetailResponseBodyWorkflowTaskDetailWorkflow extends $dara.Model {
    static names() {
        return {
            appId: 'AppId',
            createTime: 'CreateTime',
            modifiedTime: 'ModifiedTime',
            name: 'Name',
            status: 'Status',
            type: 'Type',
            workflowId: 'WorkflowId',
        };
    }
    static types() {
        return {
            appId: 'string',
            createTime: 'string',
            modifiedTime: 'string',
            name: 'string',
            status: 'string',
            type: 'string',
            workflowId: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetJobDetailResponseBodyWorkflowTaskDetailWorkflow = GetJobDetailResponseBodyWorkflowTaskDetailWorkflow;
class GetJobDetailResponseBodyWorkflowTaskDetail extends $dara.Model {
    static names() {
        return {
            activityResults: 'ActivityResults',
            createTime: 'CreateTime',
            finishTime: 'FinishTime',
            status: 'Status',
            taskId: 'TaskId',
            taskInput: 'TaskInput',
            userData: 'UserData',
            workflow: 'Workflow',
        };
    }
    static types() {
        return {
            activityResults: 'string',
            createTime: 'string',
            finishTime: 'string',
            status: 'string',
            taskId: 'string',
            taskInput: 'string',
            userData: 'string',
            workflow: GetJobDetailResponseBodyWorkflowTaskDetailWorkflow,
        };
    }
    validate() {
        if (this.workflow && typeof this.workflow.validate === 'function') {
            this.workflow.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetJobDetailResponseBodyWorkflowTaskDetail = GetJobDetailResponseBodyWorkflowTaskDetail;
class GetJobDetailResponseBody extends $dara.Model {
    static names() {
        return {
            AIJobDetail: 'AIJobDetail',
            jobType: 'JobType',
            requestId: 'RequestId',
            snapshotJobDetail: 'SnapshotJobDetail',
            transcodeJobDetail: 'TranscodeJobDetail',
            workflowTaskDetail: 'WorkflowTaskDetail',
        };
    }
    static types() {
        return {
            AIJobDetail: GetJobDetailResponseBodyAIJobDetail,
            jobType: 'string',
            requestId: 'string',
            snapshotJobDetail: GetJobDetailResponseBodySnapshotJobDetail,
            transcodeJobDetail: GetJobDetailResponseBodyTranscodeJobDetail,
            workflowTaskDetail: GetJobDetailResponseBodyWorkflowTaskDetail,
        };
    }
    validate() {
        if (this.AIJobDetail && typeof this.AIJobDetail.validate === 'function') {
            this.AIJobDetail.validate();
        }
        if (this.snapshotJobDetail && typeof this.snapshotJobDetail.validate === 'function') {
            this.snapshotJobDetail.validate();
        }
        if (this.transcodeJobDetail && typeof this.transcodeJobDetail.validate === 'function') {
            this.transcodeJobDetail.validate();
        }
        if (this.workflowTaskDetail && typeof this.workflowTaskDetail.validate === 'function') {
            this.workflowTaskDetail.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetJobDetailResponseBody = GetJobDetailResponseBody;
