// @alicloud/vod20170321@3.11.0/dist/models/GetMediaAuditResultTimelineResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetMediaAuditResultTimelineResponseBody = exports.GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimeline = exports.GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineTerrorism = exports.GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelinePorn = exports.GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineLogo = exports.GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineLive = exports.GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineAd = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineAd extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineAd = GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineAd;
class GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineLive extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineLive = GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineLive;
class GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineLogo extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineLogo = GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineLogo;
class GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelinePorn extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelinePorn = GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelinePorn;
class GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineTerrorism extends $dara.Model {
    static names() {
        return {
            label: 'Label',
            score: 'Score',
            timestamp: 'Timestamp',
        };
    }
    static types() {
        return {
            label: 'string',
            score: 'string',
            timestamp: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineTerrorism = GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineTerrorism;
class GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimeline extends $dara.Model {
    static names() {
        return {
            ad: 'Ad',
            live: 'Live',
            logo: 'Logo',
            porn: 'Porn',
            terrorism: 'Terrorism',
        };
    }
    static types() {
        return {
            ad: { 'type': 'array', 'itemType': GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineAd },
            live: { 'type': 'array', 'itemType': GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineLive },
            logo: { 'type': 'array', 'itemType': GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineLogo },
            porn: { 'type': 'array', 'itemType': GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelinePorn },
            terrorism: { 'type': 'array', 'itemType': GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimelineTerrorism },
        };
    }
    validate() {
        if (Array.isArray(this.ad)) {
            $dara.Model.validateArray(this.ad);
        }
        if (Array.isArray(this.live)) {
            $dara.Model.validateArray(this.live);
        }
        if (Array.isArray(this.logo)) {
            $dara.Model.validateArray(this.logo);
        }
        if (Array.isArray(this.porn)) {
            $dara.Model.validateArray(this.porn);
        }
        if (Array.isArray(this.terrorism)) {
            $dara.Model.validateArray(this.terrorism);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimeline = GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimeline;
class GetMediaAuditResultTimelineResponseBody extends $dara.Model {
    static names() {
        return {
            mediaAuditResultTimeline: 'MediaAuditResultTimeline',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            mediaAuditResultTimeline: GetMediaAuditResultTimelineResponseBodyMediaAuditResultTimeline,
            requestId: 'string',
        };
    }
    validate() {
        if (this.mediaAuditResultTimeline && typeof this.mediaAuditResultTimeline.validate === 'function') {
            this.mediaAuditResultTimeline.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaAuditResultTimelineResponseBody = GetMediaAuditResultTimelineResponseBody;
