// @alicloud/vod20170321@3.11.0/dist/models/GetMediaDnaresultResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetMediaDNAResultResponseBody = exports.GetMediaDNAResultResponseBodyDNAResult = exports.GetMediaDNAResultResponseBodyDNAResultVideoDNA = exports.GetMediaDNAResultResponseBodyDNAResultVideoDNADetail = exports.GetMediaDNAResultResponseBodyDNAResultVideoDNADetailInput = exports.GetMediaDNAResultResponseBodyDNAResultVideoDNADetailDuplication = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetMediaDNAResultResponseBodyDNAResultVideoDNADetailDuplication extends $dara.Model {
    static names() {
        return {
            duration: 'Duration',
            start: 'Start',
        };
    }
    static types() {
        return {
            duration: 'string',
            start: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaDNAResultResponseBodyDNAResultVideoDNADetailDuplication = GetMediaDNAResultResponseBodyDNAResultVideoDNADetailDuplication;
class GetMediaDNAResultResponseBodyDNAResultVideoDNADetailInput extends $dara.Model {
    static names() {
        return {
            duration: 'Duration',
            start: 'Start',
        };
    }
    static types() {
        return {
            duration: 'string',
            start: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaDNAResultResponseBodyDNAResultVideoDNADetailInput = GetMediaDNAResultResponseBodyDNAResultVideoDNADetailInput;
class GetMediaDNAResultResponseBodyDNAResultVideoDNADetail extends $dara.Model {
    static names() {
        return {
            duplication: 'Duplication',
            input: 'Input',
        };
    }
    static types() {
        return {
            duplication: GetMediaDNAResultResponseBodyDNAResultVideoDNADetailDuplication,
            input: GetMediaDNAResultResponseBodyDNAResultVideoDNADetailInput,
        };
    }
    validate() {
        if (this.duplication && typeof this.duplication.validate === 'function') {
            this.duplication.validate();
        }
        if (this.input && typeof this.input.validate === 'function') {
            this.input.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaDNAResultResponseBodyDNAResultVideoDNADetail = GetMediaDNAResultResponseBodyDNAResultVideoDNADetail;
class GetMediaDNAResultResponseBodyDNAResultVideoDNA extends $dara.Model {
    static names() {
        return {
            detail: 'Detail',
            primaryKey: 'PrimaryKey',
            similarity: 'Similarity',
        };
    }
    static types() {
        return {
            detail: { 'type': 'array', 'itemType': GetMediaDNAResultResponseBodyDNAResultVideoDNADetail },
            primaryKey: 'string',
            similarity: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.detail)) {
            $dara.Model.validateArray(this.detail);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaDNAResultResponseBodyDNAResultVideoDNA = GetMediaDNAResultResponseBodyDNAResultVideoDNA;
class GetMediaDNAResultResponseBodyDNAResult extends $dara.Model {
    static names() {
        return {
            videoDNA: 'VideoDNA',
        };
    }
    static types() {
        return {
            videoDNA: { 'type': 'array', 'itemType': GetMediaDNAResultResponseBodyDNAResultVideoDNA },
        };
    }
    validate() {
        if (Array.isArray(this.videoDNA)) {
            $dara.Model.validateArray(this.videoDNA);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaDNAResultResponseBodyDNAResult = GetMediaDNAResultResponseBodyDNAResult;
class GetMediaDNAResultResponseBody extends $dara.Model {
    static names() {
        return {
            DNAResult: 'DNAResult',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            DNAResult: GetMediaDNAResultResponseBodyDNAResult,
            requestId: 'string',
        };
    }
    validate() {
        if (this.DNAResult && typeof this.DNAResult.validate === 'function') {
            this.DNAResult.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetMediaDNAResultResponseBody = GetMediaDNAResultResponseBody;
