// @alicloud/vod20170321@3.11.0/dist/models/GetPlayInfoResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetPlayInfoResponseBody = exports.GetPlayInfoResponseBodyVideoBase = exports.GetPlayInfoResponseBodyPlayInfoList = exports.GetPlayInfoResponseBodyPlayInfoListPlayInfo = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetPlayInfoResponseBodyPlayInfoListPlayInfo extends $dara.Model {
    static names() {
        return {
            bitDepth: 'BitDepth',
            bitrate: 'Bitrate',
            codecName: 'CodecName',
            creationTime: 'CreationTime',
            definition: 'Definition',
            duration: 'Duration',
            encrypt: 'Encrypt',
            encryptMode: 'EncryptMode',
            encryptType: 'EncryptType',
            format: 'Format',
            fps: 'Fps',
            HDRType: 'HDRType',
            height: 'Height',
            jobExt: 'JobExt',
            jobId: 'JobId',
            jobType: 'JobType',
            modificationTime: 'ModificationTime',
            narrowBandType: 'NarrowBandType',
            playURL: 'PlayURL',
            size: 'Size',
            specification: 'Specification',
            status: 'Status',
            streamType: 'StreamType',
            watermarkId: 'WatermarkId',
            width: 'Width',
        };
    }
    static types() {
        return {
            bitDepth: 'number',
            bitrate: 'string',
            codecName: 'string',
            creationTime: 'string',
            definition: 'string',
            duration: 'string',
            encrypt: 'number',
            encryptMode: 'string',
            encryptType: 'string',
            format: 'string',
            fps: 'string',
            HDRType: 'string',
            height: 'number',
            jobExt: 'string',
            jobId: 'string',
            jobType: 'number',
            modificationTime: 'string',
            narrowBandType: 'string',
            playURL: 'string',
            size: 'number',
            specification: 'string',
            status: 'string',
            streamType: 'string',
            watermarkId: 'string',
            width: 'number',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPlayInfoResponseBodyPlayInfoListPlayInfo = GetPlayInfoResponseBodyPlayInfoListPlayInfo;
class GetPlayInfoResponseBodyPlayInfoList extends $dara.Model {
    static names() {
        return {
            playInfo: 'PlayInfo',
        };
    }
    static types() {
        return {
            playInfo: { 'type': 'array', 'itemType': GetPlayInfoResponseBodyPlayInfoListPlayInfo },
        };
    }
    validate() {
        if (Array.isArray(this.playInfo)) {
            $dara.Model.validateArray(this.playInfo);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPlayInfoResponseBodyPlayInfoList = GetPlayInfoResponseBodyPlayInfoList;
class GetPlayInfoResponseBodyVideoBase extends $dara.Model {
    static names() {
        return {
            coverURL: 'CoverURL',
            creationTime: 'CreationTime',
            danMuURL: 'DanMuURL',
            duration: 'Duration',
            mediaType: 'MediaType',
            status: 'Status',
            storageClass: 'StorageClass',
            title: 'Title',
            videoId: 'VideoId',
        };
    }
    static types() {
        return {
            coverURL: 'string',
            creationTime: 'string',
            danMuURL: 'string',
            duration: 'string',
            mediaType: 'string',
            status: 'string',
            storageClass: 'string',
            title: 'string',
            videoId: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPlayInfoResponseBodyVideoBase = GetPlayInfoResponseBodyVideoBase;
class GetPlayInfoResponseBody extends $dara.Model {
    static names() {
        return {
            playInfoList: 'PlayInfoList',
            requestId: 'RequestId',
            videoBase: 'VideoBase',
        };
    }
    static types() {
        return {
            playInfoList: GetPlayInfoResponseBodyPlayInfoList,
            requestId: 'string',
            videoBase: GetPlayInfoResponseBodyVideoBase,
        };
    }
    validate() {
        if (this.playInfoList && typeof this.playInfoList.validate === 'function') {
            this.playInfoList.validate();
        }
        if (this.videoBase && typeof this.videoBase.validate === 'function') {
            this.videoBase.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetPlayInfoResponseBody = GetPlayInfoResponseBody;
