// @alicloud/vod20170321@3.11.0/dist/models/GetTranscodeSummaryResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetTranscodeSummaryResponseBody = exports.GetTranscodeSummaryResponseBodyTranscodeSummaryList = exports.GetTranscodeSummaryResponseBodyTranscodeSummaryListTranscodeJobInfoSummaryList = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetTranscodeSummaryResponseBodyTranscodeSummaryListTranscodeJobInfoSummaryList extends $dara.Model {
    static names() {
        return {
            bitrate: 'Bitrate',
            completeTime: 'CompleteTime',
            creationTime: 'CreationTime',
            duration: 'Duration',
            errorCode: 'ErrorCode',
            errorMessage: 'ErrorMessage',
            filesize: 'Filesize',
            format: 'Format',
            fps: 'Fps',
            height: 'Height',
            transcodeJobStatus: 'TranscodeJobStatus',
            transcodeProgress: 'TranscodeProgress',
            transcodeTemplateId: 'TranscodeTemplateId',
            watermarkIdList: 'WatermarkIdList',
            width: 'Width',
        };
    }
    static types() {
        return {
            bitrate: 'string',
            completeTime: 'string',
            creationTime: 'string',
            duration: 'string',
            errorCode: 'string',
            errorMessage: 'string',
            filesize: 'number',
            format: 'string',
            fps: 'string',
            height: 'string',
            transcodeJobStatus: 'string',
            transcodeProgress: 'number',
            transcodeTemplateId: 'string',
            watermarkIdList: { 'type': 'array', 'itemType': 'string' },
            width: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.watermarkIdList)) {
            $dara.Model.validateArray(this.watermarkIdList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscodeSummaryResponseBodyTranscodeSummaryListTranscodeJobInfoSummaryList = GetTranscodeSummaryResponseBodyTranscodeSummaryListTranscodeJobInfoSummaryList;
class GetTranscodeSummaryResponseBodyTranscodeSummaryList extends $dara.Model {
    static names() {
        return {
            completeTime: 'CompleteTime',
            creationTime: 'CreationTime',
            transcodeJobInfoSummaryList: 'TranscodeJobInfoSummaryList',
            transcodeStatus: 'TranscodeStatus',
            transcodeTemplateGroupId: 'TranscodeTemplateGroupId',
            videoId: 'VideoId',
        };
    }
    static types() {
        return {
            completeTime: 'string',
            creationTime: 'string',
            transcodeJobInfoSummaryList: { 'type': 'array', 'itemType': GetTranscodeSummaryResponseBodyTranscodeSummaryListTranscodeJobInfoSummaryList },
            transcodeStatus: 'string',
            transcodeTemplateGroupId: 'string',
            videoId: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.transcodeJobInfoSummaryList)) {
            $dara.Model.validateArray(this.transcodeJobInfoSummaryList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscodeSummaryResponseBodyTranscodeSummaryList = GetTranscodeSummaryResponseBodyTranscodeSummaryList;
class GetTranscodeSummaryResponseBody extends $dara.Model {
    static names() {
        return {
            nonExistVideoIds: 'NonExistVideoIds',
            requestId: 'RequestId',
            transcodeSummaryList: 'TranscodeSummaryList',
        };
    }
    static types() {
        return {
            nonExistVideoIds: { 'type': 'array', 'itemType': 'string' },
            requestId: 'string',
            transcodeSummaryList: { 'type': 'array', 'itemType': GetTranscodeSummaryResponseBodyTranscodeSummaryList },
        };
    }
    validate() {
        if (Array.isArray(this.nonExistVideoIds)) {
            $dara.Model.validateArray(this.nonExistVideoIds);
        }
        if (Array.isArray(this.transcodeSummaryList)) {
            $dara.Model.validateArray(this.transcodeSummaryList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetTranscodeSummaryResponseBody = GetTranscodeSummaryResponseBody;
