// @alicloud/vod20170321@3.11.0/dist/models/GetUploadDetailsResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetUploadDetailsResponseBody = exports.GetUploadDetailsResponseBodyUploadDetails = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetUploadDetailsResponseBodyUploadDetails extends $dara.Model {
    static names() {
        return {
            completionTime: 'CompletionTime',
            creationTime: 'CreationTime',
            deviceModel: 'DeviceModel',
            fileSize: 'FileSize',
            mediaId: 'MediaId',
            modificationTime: 'ModificationTime',
            status: 'Status',
            title: 'Title',
            uploadIP: 'UploadIP',
            uploadRatio: 'UploadRatio',
            uploadSize: 'UploadSize',
            uploadSource: 'UploadSource',
            uploadStatus: 'UploadStatus',
        };
    }
    static types() {
        return {
            completionTime: 'string',
            creationTime: 'string',
            deviceModel: 'string',
            fileSize: 'number',
            mediaId: 'string',
            modificationTime: 'string',
            status: 'string',
            title: 'string',
            uploadIP: 'string',
            uploadRatio: 'number',
            uploadSize: 'number',
            uploadSource: 'string',
            uploadStatus: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUploadDetailsResponseBodyUploadDetails = GetUploadDetailsResponseBodyUploadDetails;
class GetUploadDetailsResponseBody extends $dara.Model {
    static names() {
        return {
            forbiddenMediaIds: 'ForbiddenMediaIds',
            nonExistMediaIds: 'NonExistMediaIds',
            requestId: 'RequestId',
            uploadDetails: 'UploadDetails',
        };
    }
    static types() {
        return {
            forbiddenMediaIds: { 'type': 'array', 'itemType': 'string' },
            nonExistMediaIds: { 'type': 'array', 'itemType': 'string' },
            requestId: 'string',
            uploadDetails: { 'type': 'array', 'itemType': GetUploadDetailsResponseBodyUploadDetails },
        };
    }
    validate() {
        if (Array.isArray(this.forbiddenMediaIds)) {
            $dara.Model.validateArray(this.forbiddenMediaIds);
        }
        if (Array.isArray(this.nonExistMediaIds)) {
            $dara.Model.validateArray(this.nonExistMediaIds);
        }
        if (Array.isArray(this.uploadDetails)) {
            $dara.Model.validateArray(this.uploadDetails);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetUploadDetailsResponseBody = GetUploadDetailsResponseBody;
