// @alicloud/vod20170321@3.11.0/dist/models/GetVideoInfoResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetVideoInfoResponseBody = exports.GetVideoInfoResponseBodyVideo = exports.GetVideoInfoResponseBodyVideoSnapshots = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetVideoInfoResponseBodyVideoSnapshots extends $dara.Model {
    static names() {
        return {
            snapshot: 'Snapshot',
        };
    }
    static types() {
        return {
            snapshot: { 'type': 'array', 'itemType': 'string' },
        };
    }
    validate() {
        if (Array.isArray(this.snapshot)) {
            $dara.Model.validateArray(this.snapshot);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetVideoInfoResponseBodyVideoSnapshots = GetVideoInfoResponseBodyVideoSnapshots;
class GetVideoInfoResponseBodyVideo extends $dara.Model {
    static names() {
        return {
            appId: 'AppId',
            auditStatus: 'AuditStatus',
            cateId: 'CateId',
            cateName: 'CateName',
            coverURL: 'CoverURL',
            creationTime: 'CreationTime',
            customMediaInfo: 'CustomMediaInfo',
            description: 'Description',
            downloadSwitch: 'DownloadSwitch',
            duration: 'Duration',
            modificationTime: 'ModificationTime',
            referenceId: 'ReferenceId',
            regionId: 'RegionId',
            restoreExpiration: 'RestoreExpiration',
            restoreStatus: 'RestoreStatus',
            size: 'Size',
            snapshots: 'Snapshots',
            status: 'Status',
            storageClass: 'StorageClass',
            storageLocation: 'StorageLocation',
            tags: 'Tags',
            templateGroupId: 'TemplateGroupId',
            title: 'Title',
            userData: 'UserData',
            videoId: 'VideoId',
        };
    }
    static types() {
        return {
            appId: 'string',
            auditStatus: 'string',
            cateId: 'number',
            cateName: 'string',
            coverURL: 'string',
            creationTime: 'string',
            customMediaInfo: 'string',
            description: 'string',
            downloadSwitch: 'string',
            duration: 'number',
            modificationTime: 'string',
            referenceId: 'string',
            regionId: 'string',
            restoreExpiration: 'string',
            restoreStatus: 'string',
            size: 'number',
            snapshots: GetVideoInfoResponseBodyVideoSnapshots,
            status: 'string',
            storageClass: 'string',
            storageLocation: 'string',
            tags: 'string',
            templateGroupId: 'string',
            title: 'string',
            userData: 'string',
            videoId: 'string',
        };
    }
    validate() {
        if (this.snapshots && typeof this.snapshots.validate === 'function') {
            this.snapshots.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetVideoInfoResponseBodyVideo = GetVideoInfoResponseBodyVideo;
class GetVideoInfoResponseBody extends $dara.Model {
    static names() {
        return {
            requestId: 'RequestId',
            video: 'Video',
        };
    }
    static types() {
        return {
            requestId: 'string',
            video: GetVideoInfoResponseBodyVideo,
        };
    }
    validate() {
        if (this.video && typeof this.video.validate === 'function') {
            this.video.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetVideoInfoResponseBody = GetVideoInfoResponseBody;
