// @alicloud/vod20170321@3.11.0/dist/models/ListAijobResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAIJobResponseBody = exports.ListAIJobResponseBodyNonExistAIJobIds = exports.ListAIJobResponseBodyAIJobList = exports.ListAIJobResponseBodyAIJobListAIJob = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class ListAIJobResponseBodyAIJobListAIJob extends $dara.Model {
    static names() {
        return {
            code: 'Code',
            completeTime: 'CompleteTime',
            creationTime: 'CreationTime',
            data: 'Data',
            jobId: 'JobId',
            mediaId: 'MediaId',
            message: 'Message',
            status: 'Status',
            type: 'Type',
        };
    }
    static types() {
        return {
            code: 'string',
            completeTime: 'string',
            creationTime: 'string',
            data: 'string',
            jobId: 'string',
            mediaId: 'string',
            message: 'string',
            status: 'string',
            type: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAIJobResponseBodyAIJobListAIJob = ListAIJobResponseBodyAIJobListAIJob;
class ListAIJobResponseBodyAIJobList extends $dara.Model {
    static names() {
        return {
            AIJob: 'AIJob',
        };
    }
    static types() {
        return {
            AIJob: { 'type': 'array', 'itemType': ListAIJobResponseBodyAIJobListAIJob },
        };
    }
    validate() {
        if (Array.isArray(this.AIJob)) {
            $dara.Model.validateArray(this.AIJob);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAIJobResponseBodyAIJobList = ListAIJobResponseBodyAIJobList;
class ListAIJobResponseBodyNonExistAIJobIds extends $dara.Model {
    static names() {
        return {
            string: 'String',
        };
    }
    static types() {
        return {
            string: { 'type': 'array', 'itemType': 'string' },
        };
    }
    validate() {
        if (Array.isArray(this.string)) {
            $dara.Model.validateArray(this.string);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAIJobResponseBodyNonExistAIJobIds = ListAIJobResponseBodyNonExistAIJobIds;
class ListAIJobResponseBody extends $dara.Model {
    static names() {
        return {
            AIJobList: 'AIJobList',
            nonExistAIJobIds: 'NonExistAIJobIds',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            AIJobList: ListAIJobResponseBodyAIJobList,
            nonExistAIJobIds: ListAIJobResponseBodyNonExistAIJobIds,
            requestId: 'string',
        };
    }
    validate() {
        if (this.AIJobList && typeof this.AIJobList.validate === 'function') {
            this.AIJobList.validate();
        }
        if (this.nonExistAIJobIds && typeof this.nonExistAIJobIds.validate === 'function') {
            this.nonExistAIJobIds.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.ListAIJobResponseBody = ListAIJobResponseBody;
