// @alicloud/vod20170321@3.11.0/dist/models/SubmitAijobResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubmitAIJobResponseBody = exports.SubmitAIJobResponseBodyAIJobList = exports.SubmitAIJobResponseBodyAIJobListAIJob = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class SubmitAIJobResponseBodyAIJobListAIJob extends $dara.Model {
    static names() {
        return {
            jobId: 'JobId',
            mediaId: 'MediaId',
            type: 'Type',
        };
    }
    static types() {
        return {
            jobId: 'string',
            mediaId: 'string',
            type: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SubmitAIJobResponseBodyAIJobListAIJob = SubmitAIJobResponseBodyAIJobListAIJob;
class SubmitAIJobResponseBodyAIJobList extends $dara.Model {
    static names() {
        return {
            AIJob: 'AIJob',
        };
    }
    static types() {
        return {
            AIJob: { 'type': 'array', 'itemType': SubmitAIJobResponseBodyAIJobListAIJob },
        };
    }
    validate() {
        if (Array.isArray(this.AIJob)) {
            $dara.Model.validateArray(this.AIJob);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SubmitAIJobResponseBodyAIJobList = SubmitAIJobResponseBodyAIJobList;
class SubmitAIJobResponseBody extends $dara.Model {
    static names() {
        return {
            AIJobList: 'AIJobList',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            AIJobList: SubmitAIJobResponseBodyAIJobList,
            requestId: 'string',
        };
    }
    validate() {
        if (this.AIJobList && typeof this.AIJobList.validate === 'function') {
            this.AIJobList.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.SubmitAIJobResponseBody = SubmitAIJobResponseBody;
