// @alicloud/vod20170321@3.11.0/dist/models/UpdateMediaStorageClassResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateMediaStorageClassResponseBody = exports.UpdateMediaStorageClassResponseBodyIgnoredList = exports.UpdateMediaStorageClassResponseBodyForbiddenList = exports.UpdateMediaStorageClassResponseBodyForbiddenListMediaForbiddenReasonDTO = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class UpdateMediaStorageClassResponseBodyForbiddenListMediaForbiddenReasonDTO extends $dara.Model {
    static names() {
        return {
            mediaId: 'MediaId',
            reason: 'Reason',
        };
    }
    static types() {
        return {
            mediaId: 'string',
            reason: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMediaStorageClassResponseBodyForbiddenListMediaForbiddenReasonDTO = UpdateMediaStorageClassResponseBodyForbiddenListMediaForbiddenReasonDTO;
class UpdateMediaStorageClassResponseBodyForbiddenList extends $dara.Model {
    static names() {
        return {
            mediaForbiddenReasonDTO: 'MediaForbiddenReasonDTO',
        };
    }
    static types() {
        return {
            mediaForbiddenReasonDTO: { 'type': 'array', 'itemType': UpdateMediaStorageClassResponseBodyForbiddenListMediaForbiddenReasonDTO },
        };
    }
    validate() {
        if (Array.isArray(this.mediaForbiddenReasonDTO)) {
            $dara.Model.validateArray(this.mediaForbiddenReasonDTO);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMediaStorageClassResponseBodyForbiddenList = UpdateMediaStorageClassResponseBodyForbiddenList;
class UpdateMediaStorageClassResponseBodyIgnoredList extends $dara.Model {
    static names() {
        return {
            mediaId: 'MediaId',
        };
    }
    static types() {
        return {
            mediaId: { 'type': 'array', 'itemType': 'string' },
        };
    }
    validate() {
        if (Array.isArray(this.mediaId)) {
            $dara.Model.validateArray(this.mediaId);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMediaStorageClassResponseBodyIgnoredList = UpdateMediaStorageClassResponseBodyIgnoredList;
class UpdateMediaStorageClassResponseBody extends $dara.Model {
    static names() {
        return {
            forbiddenList: 'ForbiddenList',
            ignoredList: 'IgnoredList',
            requestId: 'RequestId',
            status: 'Status',
        };
    }
    static types() {
        return {
            forbiddenList: UpdateMediaStorageClassResponseBodyForbiddenList,
            ignoredList: UpdateMediaStorageClassResponseBodyIgnoredList,
            requestId: 'string',
            status: 'string',
        };
    }
    validate() {
        if (this.forbiddenList && typeof this.forbiddenList.validate === 'function') {
            this.forbiddenList.validate();
        }
        if (this.ignoredList && typeof this.ignoredList.validate === 'function') {
            this.ignoredList.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.UpdateMediaStorageClassResponseBody = UpdateMediaStorageClassResponseBody;
